// vcc-ui/lib/components/flex/stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Row = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = require("./");

var _default = {
  title: 'Flex'
};
exports.default = _default;
var flexStyle = {
  border: '1px solid grey',
  padding: 20,
  margin: 5
};

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Flex, {
    extend: flexStyle
  }, "Hello"), /*#__PURE__*/_react.default.createElement(_.Flex, {
    extend: flexStyle
  }, "World"));
};

exports.Default = Default;

var Row = function Row() {
  return /*#__PURE__*/_react.default.createElement(_.Flex, {
    extend: {
      flexDirection: 'row'
    }
  }, /*#__PURE__*/_react.default.createElement(_.Flex, {
    extend: flexStyle
  }, "Hello"), /*#__PURE__*/_react.default.createElement(_.Flex, {
    extend: flexStyle
  }, "World"));
};

exports.Row = Row;
