// vcc-ui/lib/components/click/stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.withExtend = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = require("./");

var _default = {
  title: 'Click'
};
exports.default = _default;

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement(_.Click, {
    onClick: function onClick() {
      alert('Hello!');
    }
  }, "This is a clickable unstyled button");
};

exports.Default = Default;

var withExtend = function withExtend() {
  return /*#__PURE__*/_react.default.createElement(_.Click, {
    extend: {
      display: 'inline-block',
      padding: '16px 24px',
      color: 'hotpink',
      background: '#f1f1f1',
      fontSize: 16,
      transition: 'all 0.2s ease-in-out 0s',
      ':hover': {
        transform: 'rotate(2deg) scale(1.1)',
        background: 'lightblue',
        color: '#fff'
      }
    },
    href: "https://www.volvocars.com/"
  }, "Custom Style Button");
};

exports.withExtend = withExtend;
