// vcc-ui/es/components/card/stories.js
import React from 'react';
import { Text, Button, Flex, Spacer } from '../../';
import { Card } from './';
import { CardContent } from '../card-content';
export default {
  title: 'Card'
};
export var Default = function Default() {
  return /*#__PURE__*/React.createElement(Card, null, /*#__PURE__*/React.createElement(CardContent, null, /*#__PURE__*/React.createElement(Text, {
    variant: "ootah"
  }, "Card Component"), /*#__PURE__*/React.createElement(Spacer, null), /*#__PURE__*/React.createElement(Text, null, "This is a card")));
};
export var Selected = function Selected() {
  return /*#__PURE__*/React.createElement(Card, {
    selected: true
  }, /*#__PURE__*/React.createElement(CardContent, null, /*#__PURE__*/React.createElement(Text, {
    variant: "ootah"
  }, "Card Component"), /*#__PURE__*/React.createElement(Spacer, null), /*#__PURE__*/React.createElement(Text, null, "This is a card is in a selected state")));
};
export var withHref = function withHref() {
  return /*#__PURE__*/React.createElement(Card, {
    href: "#test"
  }, /*#__PURE__*/React.createElement(CardContent, null, /*#__PURE__*/React.createElement(Text, {
    variant: "ootah"
  }, "Card Component"), /*#__PURE__*/React.createElement(Spacer, null), /*#__PURE__*/React.createElement(Text, null, "This is a card has a href prop and is a link")));
};
export var withNestedOnClick = function withNestedOnClick() {
  return /*#__PURE__*/React.createElement(Card, {
    onClick: function onClick() {
      alert('button clicked');
    }
  }, /*#__PURE__*/React.createElement(CardContent, null, /*#__PURE__*/React.createElement(Text, {
    variant: "ootah"
  }, "Card Component"), /*#__PURE__*/React.createElement(Spacer, null), /*#__PURE__*/React.createElement(Text, null, "This is a card has an onClick handler and contains buttons with links"), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, {
    extend: {
      flexDirection: 'row'
    }
  }, /*#__PURE__*/React.createElement(Button, {
    onClick: function onClick(e) {
      e.stopPropagation();
    },
    href: "#buy"
  }, "Buy Now"), /*#__PURE__*/React.createElement(Button, {
    onClick: function onClick(e) {
      e.stopPropagation();
    },
    href: "#learn-more",
    variant: "text"
  }, "Learn More"))));
};
export var withImage = function withImage() {
  return /*#__PURE__*/React.createElement(Card, null, /*#__PURE__*/React.createElement(Flex, {
    extend: {
      justifyContent: 'center',
      padding: 32,
      background: 'linear-gradient(pink, blue)'
    }
  }, /*#__PURE__*/React.createElement(Text, {
    variant: "bates",
    extend: {
      color: '#fff'
    }
  }, "Fake Image")), /*#__PURE__*/React.createElement(CardContent, null, /*#__PURE__*/React.createElement(Text, {
    variant: "ootah"
  }, "Card Component"), /*#__PURE__*/React.createElement(Spacer, null), /*#__PURE__*/React.createElement(Text, null, "This is a card component")));
};
export var withSelectedImageAndHref = function withSelectedImageAndHref() {
  return /*#__PURE__*/React.createElement(Card, {
    selected: true,
    href: "#test"
  }, /*#__PURE__*/React.createElement(Flex, {
    extend: {
      justifyContent: 'center',
      padding: 32,
      background: 'linear-gradient(pink, blue)'
    }
  }, /*#__PURE__*/React.createElement(Text, {
    variant: "bates",
    extend: {
      color: '#fff'
    }
  }, "Fake Image")), /*#__PURE__*/React.createElement(CardContent, null, /*#__PURE__*/React.createElement(Text, {
    variant: "ootah"
  }, "Card Component"), /*#__PURE__*/React.createElement(Spacer, null), /*#__PURE__*/React.createElement(Text, null, "This is a card component")));
};
