// vcc-ui/es/components/text/stories.js
import React from 'react';
import { Text, Spacer, Flex } from '../../';
export default {
  title: 'Text'
};
var TEXT = 'A Quick Brown Fox Jumps Over The Lazy Dog';
export var Default = function Default() {
  return /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    variant: "ootah"
  }, "Core Text Styles"), /*#__PURE__*/React.createElement(Spacer, {
    size: 4
  }), /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Headings"), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Peary"), /*#__PURE__*/React.createElement(Text, {
    variant: "peary"
  }, TEXT)), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Cook"), /*#__PURE__*/React.createElement(Text, {
    variant: "cook"
  }, TEXT)), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Ootah"), /*#__PURE__*/React.createElement(Text, {
    variant: "ootah"
  }, TEXT)), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Ootah - Emphasis"), /*#__PURE__*/React.createElement(Text, {
    variant: "ootah",
    subStyle: "emphasis"
  }, TEXT)), /*#__PURE__*/React.createElement(Spacer, {
    size: 4
  }), /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Sub-Headings"), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Hillary"), /*#__PURE__*/React.createElement(Text, {
    variant: "hillary"
  }, TEXT)), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Hillary - Inline Link"), /*#__PURE__*/React.createElement(Text, {
    variant: "hillary",
    subStyle: "inline-link"
  }, TEXT)), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Hillary - Emphasis"), /*#__PURE__*/React.createElement(Text, {
    variant: "hillary",
    subStyle: "emphasis"
  }, TEXT)), /*#__PURE__*/React.createElement(Spacer, {
    size: 4
  }), /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Body"), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Columbus"), /*#__PURE__*/React.createElement(Text, {
    variant: "columbus"
  }, TEXT)), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Columbus - Inline Link"), /*#__PURE__*/React.createElement(Text, {
    variant: "columbus",
    subStyle: "inline-link"
  }, TEXT)), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Columbus - Emphasis"), /*#__PURE__*/React.createElement(Text, {
    variant: "columbus",
    subStyle: "emphasis"
  }, TEXT)), /*#__PURE__*/React.createElement(Spacer, {
    size: 4
  }), /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Micro"), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Bates"), /*#__PURE__*/React.createElement(Text, {
    variant: "bates"
  }, TEXT)), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Bates - Inline Link"), /*#__PURE__*/React.createElement(Text, {
    variant: "bates",
    subStyle: "inline-link"
  }, TEXT)), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Bates - Emphasis"), /*#__PURE__*/React.createElement(Text, {
    variant: "bates",
    subStyle: "emphasis"
  }, TEXT)), "=======", /*#__PURE__*/React.createElement(Spacer, {
    size: 4
  }), /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Sub-Headings"), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Hillary"), /*#__PURE__*/React.createElement(Text, {
    variant: "hillary"
  }, TEXT)), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Hillary - Inline Link"), /*#__PURE__*/React.createElement(Text, {
    variant: "hillary",
    href: "",
    subStyle: "inline-link"
  }, TEXT)), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Hillary - Emphasis"), /*#__PURE__*/React.createElement(Text, {
    variant: "hillary",
    subStyle: "emphasis"
  }, TEXT)), /*#__PURE__*/React.createElement(Spacer, {
    size: 4
  }), /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Body"), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Columbus"), /*#__PURE__*/React.createElement(Text, {
    variant: "columbus"
  }, TEXT)), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Columbus - Inline Link"), /*#__PURE__*/React.createElement(Text, {
    variant: "columbus",
    href: "",
    subStyle: "inline-link"
  }, TEXT)), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Columbus - Emphasis"), /*#__PURE__*/React.createElement(Text, {
    variant: "columbus",
    subStyle: "emphasis"
  }, TEXT)), /*#__PURE__*/React.createElement(Spacer, {
    size: 4
  }), /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Micro"), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Bates"), /*#__PURE__*/React.createElement(Text, {
    variant: "bates"
  }, TEXT)), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Bates - Inline Link"), /*#__PURE__*/React.createElement(Text, {
    variant: "bates",
    href: "",
    subStyle: "inline-link"
  }, TEXT)), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Bates - Emphasis"), /*#__PURE__*/React.createElement(Text, {
    variant: "bates",
    subStyle: "emphasis"
  }, TEXT)), /*#__PURE__*/React.createElement(Spacer, {
    size: 4
  }), /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Action Text"), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Amundsen"), /*#__PURE__*/React.createElement(Text, {
    variant: "amundsen"
  }, TEXT)), /*#__PURE__*/React.createElement(Spacer, {
    size: 4
  }), /*#__PURE__*/React.createElement(Text, {
    variant: "ootah"
  }, "Outlier Text Styles"), /*#__PURE__*/React.createElement(Spacer, {
    size: 4
  }), /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Micro"), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Kelly"), /*#__PURE__*/React.createElement(Text, {
    variant: "kelly"
  }, TEXT)), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Kelly - Inline Link"), /*#__PURE__*/React.createElement(Text, {
    variant: "kelly",
    href: "",
    subStyle: "inline-link"
  }, TEXT)), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Kelly - Emphasis"), /*#__PURE__*/React.createElement(Text, {
    variant: "kelly",
    subStyle: "emphasis"
  }, TEXT)), /*#__PURE__*/React.createElement(Spacer, {
    size: 4
  }), /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Legal"), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Meir"), /*#__PURE__*/React.createElement(Text, {
    variant: "meir"
  }, TEXT)), /*#__PURE__*/React.createElement(Spacer, {
    size: 4
  }), /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Brand Statement"), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "Yang"), /*#__PURE__*/React.createElement(Text, {
    as: "h1",
    variant: "yang"
  }, TEXT)));
};
export var Nested = function Nested() {
  return /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    variant: "hillary"
  }, "Hillary with", ' ', /*#__PURE__*/React.createElement(Text, {
    variant: "ootah",
    subStyle: "emphasis"
  }, "nested Ootah emphasis")), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Text, {
    variant: "cook"
  }, "Hillary with ", /*#__PURE__*/React.createElement(Text, {
    extend: {
      fontStyle: 'italic'
    }
  }, "nested extends")), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Text, {
    variant: "ootah",
    subStyle: "emphasis"
  }, "Ootah emphasis with ", /*#__PURE__*/React.createElement(Text, {
    subStyle: "standard"
  }, "nested standard")), /*#__PURE__*/React.createElement(Spacer, {
    size: 2
  }), /*#__PURE__*/React.createElement(Text, null, "Default (Columbus) with", /*#__PURE__*/React.createElement(Text, {
    subStyle: "emphasis"
  }, "nested emphasis"), "and an", ' ', /*#__PURE__*/React.createElement(Text, {
    subStyle: "inline-link",
    href: "#"
  }, "inline link")));
};
export var Responsive = function Responsive() {
  return /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Text, {
    variant: ['hillary', 'columbus'],
    subStyle: ['standard', 'emphasis', 'standard', 'emphasis']
  }, "Responsive Text"));
};
