// vcc-ui/es/components/grid/stories.js
import _extends from "@babel/runtime/helpers/extends";
import React from 'react';
import { Grid } from './';
import { Row } from '../row';
import { Col } from '../col';
import { DebugGrid } from '../debug-grid';
export default {
  title: 'Grid'
};

var DebugBorder = function DebugBorder(props) {
  return /*#__PURE__*/React.createElement("div", _extends({
    style: {
      border: '1px solid grey',
      minHeight: '40px',
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      margin: '5px 0'
    }
  }, props));
};

export var Default = function Default() {
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(DebugGrid, {
    storybook: true
  }), /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement("h2", null, "Center"), /*#__PURE__*/React.createElement(Row, {
    align: "center"
  }, /*#__PURE__*/React.createElement(Col, {
    size: 2
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      backgroundColor: 'red'
    }
  }, "size=2")), /*#__PURE__*/React.createElement(Col, {
    size: 1
  }, /*#__PURE__*/React.createElement(DebugBorder, null, "size=1 (default)")), /*#__PURE__*/React.createElement(Col, {
    size: 7
  }, /*#__PURE__*/React.createElement(DebugBorder, null, "size=7"))), /*#__PURE__*/React.createElement(Row, {
    align: "center"
  }, /*#__PURE__*/React.createElement(Col, {
    size: 2
  }, /*#__PURE__*/React.createElement(DebugBorder, null, "size=2")), /*#__PURE__*/React.createElement(Col, {
    size: 1
  }, /*#__PURE__*/React.createElement(DebugBorder, null, "size=1 (default)")), /*#__PURE__*/React.createElement(Col, {
    size: 6
  }, /*#__PURE__*/React.createElement(DebugBorder, null, "size=6")), /*#__PURE__*/React.createElement(Col, {
    size: 1
  })), /*#__PURE__*/React.createElement(Row, {
    align: "center"
  }, /*#__PURE__*/React.createElement(Col, {
    size: 1
  }), /*#__PURE__*/React.createElement(Col, {
    size: 2
  }, /*#__PURE__*/React.createElement(DebugBorder, null, "size=2")), /*#__PURE__*/React.createElement(Col, {
    size: 1
  }, /*#__PURE__*/React.createElement(DebugBorder, null, "size=1 (default)")), /*#__PURE__*/React.createElement(Col, {
    size: 6
  }, /*#__PURE__*/React.createElement(DebugBorder, null, "size=6"))), /*#__PURE__*/React.createElement("h2", null, "Start"), /*#__PURE__*/React.createElement(Row, {
    align: "start"
  }, /*#__PURE__*/React.createElement(Col, {
    size: 2
  }, /*#__PURE__*/React.createElement(DebugBorder, null, "size=2")), /*#__PURE__*/React.createElement(Col, {
    size: 1
  }, /*#__PURE__*/React.createElement(DebugBorder, null, "size=1 (default)")), /*#__PURE__*/React.createElement(Col, {
    size: 7
  }, /*#__PURE__*/React.createElement(DebugBorder, null, "size=7"))), /*#__PURE__*/React.createElement("h2", null, "End"), /*#__PURE__*/React.createElement(Row, {
    align: "end"
  }, /*#__PURE__*/React.createElement(Col, {
    size: 2
  }, /*#__PURE__*/React.createElement(DebugBorder, null, "size=2")), /*#__PURE__*/React.createElement(Col, {
    size: 1
  }, /*#__PURE__*/React.createElement(DebugBorder, null, "size=1 (default)")), /*#__PURE__*/React.createElement(Col, {
    size: 7
  }, /*#__PURE__*/React.createElement(DebugBorder, null, "size=7")))));
};
export var NestedCols = function NestedCols() {
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(DebugGrid, {
    storybook: true
  }), /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
    size: 3
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      backgroundColor: 'grey'
    }
  }, "size=3")), /*#__PURE__*/React.createElement(Col, {
    size: 7
  }, /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
    size: 3
  }, /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
    size: 1
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      backgroundColor: 'blue'
    }
  }, "size=1")), /*#__PURE__*/React.createElement(Col, {
    size: 2
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      backgroundColor: 'green'
    }
  }, "size=2")))), /*#__PURE__*/React.createElement(Col, {
    size: 4
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      backgroundColor: 'green'
    }
  }, "size=4"))), /*#__PURE__*/React.createElement("br", null), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
    size: 5
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      backgroundColor: 'blue'
    }
  }, "size=5")), /*#__PURE__*/React.createElement(Col, {
    size: 1
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      backgroundColor: 'green'
    }
  }, "size=1")))))));
};
export var Responsive = function Responsive() {
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(DebugGrid, {
    storybook: true
  }), /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
    size: 1
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      backgroundColor: 'purple'
    }
  }, "Hello world"))), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, {
    size: 2
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      backgroundColor: 'blue'
    }
  }, "Hello world")), /*#__PURE__*/React.createElement(Col, {
    size: 2
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      backgroundColor: 'blue'
    }
  }, "Hello world")), /*#__PURE__*/React.createElement(Col, {
    size: 2
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      backgroundColor: 'blue'
    }
  }, "Hello world")), /*#__PURE__*/React.createElement(Col, {
    size: 2
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      backgroundColor: 'blue'
    }
  }, "Hello world"))), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Col, null, /*#__PURE__*/React.createElement("div", {
    style: {
      backgroundColor: 'green'
    }
  }, "I am full width always"))), /*#__PURE__*/React.createElement(Row, {
    align: "center"
  }, /*#__PURE__*/React.createElement(Col, {
    size: {
      default: 4,
      '@media (min-width: 480px)': 3,
      '@media (min-width: 1024px)': 6,
      '@media (min-width: 1600px)': 10
    }
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      backgroundColor: 'red'
    }
  }, "My size responds to your viewport width"))), /*#__PURE__*/React.createElement(Row, {
    align: "center"
  }, /*#__PURE__*/React.createElement(Col, {
    size: {
      default: 4,
      fromM: 3,
      fromL: 6,
      fromXL: 10
    }
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      backgroundColor: 'red'
    }
  }, "My size responds to your viewport width using named keys")))));
};
