// vcc-ui/lib/components/button/stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Sizes = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = require("./");

var _text = require("../text");

var _view = require("../view");

var _default = {
  title: 'Button'
};
exports.default = _default;

var DisplayWrapper = function DisplayWrapper(_ref) {
  var name = _ref.name,
      children = _ref.children;
  return /*#__PURE__*/_react.default.createElement(_view.View, {
    spacing: 1,
    extend: {
      minWidth: 250
    }
  }, /*#__PURE__*/_react.default.createElement(_text.Text, null, name), children);
};

var Row = function Row(_ref2) {
  var children = _ref2.children;
  return /*#__PURE__*/_react.default.createElement(_view.View, {
    direction: ['column', 'row'],
    spacing: 4
  }, children);
};

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement(_view.View, {
    spacing: 4
  }, /*#__PURE__*/_react.default.createElement(Row, null, /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Primary - Default"
  }, /*#__PURE__*/_react.default.createElement(_.Button, null, "Button")), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Primary - Disabled"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    disabled: true
  }, "Button")), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Primary - Loading"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    loading: true
  }, "Button"))), /*#__PURE__*/_react.default.createElement(Row, null, /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Primary/Outline - Default"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    variant: "outline"
  }, "Button")), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Primary/Outline - Disabled"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    variant: "outline",
    disabled: true
  }, "Button")), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Primary/Outline - Loading"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    variant: "outline",
    loading: true
  }, "Button"))), /*#__PURE__*/_react.default.createElement(Row, null, /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Primary/Text - Default"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    variant: "text"
  }, "Button")), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Primary/Text - Disabled"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    variant: "text",
    disabled: true
  }, "Button")), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Primary/Text - Loading"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    variant: "text",
    loading: true
  }, "Button"))), /*#__PURE__*/_react.default.createElement(Row, null, /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Secondary - Default"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    intent: "secondary"
  }, "Button")), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Secondary - Disabled"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    intent: "secondary",
    disabled: true
  }, "Button")), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Secondary - Loading"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    intent: "secondary",
    loading: true
  }, "Button"))), /*#__PURE__*/_react.default.createElement(Row, null, /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Secondary/Outline - Default"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    intent: "secondary",
    variant: "outline"
  }, "Button")), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Secondary/Outline - Disabled"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    intent: "secondary",
    variant: "outline",
    disabled: true
  }, "Button")), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Secondary/Outline - Loading"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    intent: "secondary",
    variant: "outline",
    loading: true
  }, "Button"))), /*#__PURE__*/_react.default.createElement(Row, null, /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Secondary/Text - Default"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    intent: "secondary",
    variant: "text"
  }, "Button")), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Secondary/Text - Disabled"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    intent: "secondary",
    variant: "text",
    disabled: true
  }, "Button")), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Secondary/Text - Loading"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    intent: "secondary",
    variant: "text",
    loading: true
  }, "Button"))), /*#__PURE__*/_react.default.createElement(Row, null, /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Destructive - Default"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    intent: "destructive"
  }, "Button")), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Destructive - Disabled"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    intent: "destructive",
    disabled: true
  }, "Button")), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Destructive - Loading"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    intent: "destructive",
    loading: true
  }, "Button"))), /*#__PURE__*/_react.default.createElement(Row, null, /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Destructive/Outline - Default"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    intent: "destructive",
    variant: "outline"
  }, "Button")), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Destructive/Outline - Disabled"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    intent: "destructive",
    variant: "outline",
    disabled: true
  }, "Button")), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Destructive/Outline - Loading"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    intent: "destructive",
    variant: "outline",
    loading: true
  }, "Button"))), /*#__PURE__*/_react.default.createElement(Row, null, /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Destructive/Text - Default"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    intent: "destructive",
    variant: "text"
  }, "Button")), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Destructive/Outline - Disabled"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    intent: "destructive",
    variant: "text",
    disabled: true
  }, "Button")), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Destructive/Outline - Loading"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    intent: "destructive",
    variant: "text",
    loading: true
  }, "Button"))));
};

exports.Default = Default;

var Sizes = function Sizes() {
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(Row, null, /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Small"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    size: "small"
  }, "Button")), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Medium"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    size: "medium"
  }, "Button")), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Large"
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    size: "large"
  }, "Button"))));
};

exports.Sizes = Sizes;
