// vcc-ui/lib/components/debug-grid/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DebugGrid = DebugGrid;

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _reactFela = require("react-fela");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _flex = require("../flex");

var _text = require("../text");

// Testing Only
function DebugColumn(_ref) {
  var _ref$size = _ref.size,
      size = _ref$size === void 0 ? 1 : _ref$size,
      children = _ref.children;

  var _useFela = (0, _reactFela.useFela)(),
      theme = _useFela.theme;

  return /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: (0, _defineProperty2.default)({
      boxSizing: 'border-box',
      backgroundColor: 'rgba(0, 0,0,0.1)',
      width: 1024 / 4 * size,
      marginRight: 16,
      height: '100vh',
      ':last-child': {
        marginRight: 0
      }
    }, theme.breakpoints.fromL, {
      width: 1368 / 12 * size,
      marginRight: 24,
      ':last-child': {
        marginRight: 0
      }
    })
  }, /*#__PURE__*/_react.default.createElement(_text.Text, {
    extend: {
      paddingLeft: 4
    },
    variant: "bates",
    subStyle: "emphasis"
  }, children));
}

function DebugGrid(_ref3) {
  var _ref5;

  var _ref3$storybook = _ref3.storybook,
      storybook = _ref3$storybook === void 0 ? false : _ref3$storybook,
      _ref3$disabled = _ref3.disabled,
      disabled = _ref3$disabled === void 0 ? false : _ref3$disabled;

  var _useFela2 = (0, _reactFela.useFela)(),
      theme = _useFela2.theme;

  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      display: disabled ? 'none' : 'block',
      position: 'fixed',
      zIndex: 10000,
      pointerEvents: 'none',
      opacity: 0.5,
      top: 0,
      left: 0,
      right: 0,
      bottom: 0,
      padding: storybook ? '0 24px' : 0,
      boxSizing: 'border-box'
    }
  }, /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: (0, _defineProperty2.default)({
      flexDirection: 'row',
      maxWidth: 1400,
      height: '100vh',
      margin: '0 auto',
      width: '100%',
      paddingLeft: 40,
      paddingRight: 40,
      boxSizing: 'border-box',
      backgroundColor: 'rgba(0, 0,0, 0.05)',
      display: 'none'
    }, theme.breakpoints.fromL, {
      display: 'flex'
    })
  }, (0, _toConsumableArray2.default)(Array(12).keys()).map(function (i) {
    return /*#__PURE__*/_react.default.createElement(DebugColumn, {
      key: i
    }, i + 1);
  })), /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: (_ref5 = {
      flexDirection: 'row',
      height: '100vh',
      margin: '0 auto',
      width: '100%',
      paddingLeft: 16,
      paddingRight: 16,
      boxSizing: 'border-box',
      backgroundColor: 'rgba(0, 0,0, 0.05)',
      display: 'flex'
    }, (0, _defineProperty2.default)(_ref5, theme.breakpoints.fromM, {
      paddingLeft: 24,
      paddingRight: 24
    }), (0, _defineProperty2.default)(_ref5, theme.breakpoints.fromL, {
      display: 'none'
    }), _ref5)
  }, (0, _toConsumableArray2.default)(Array(4).keys()).map(function (i) {
    return /*#__PURE__*/_react.default.createElement(DebugColumn, {
      key: i
    }, i + 1);
  })));
}

DebugGrid.propTypes = {
  /** Disable debug grid */
  disabled: _propTypes.default.bool,

  /** Enable if using in a storybook environment */
  storybook: _propTypes.default.bool
};
