// vcc-ui/lib/components/spinner/stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Thickness = exports.Multiple = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = require("./");

var _default = {
  title: 'Spinner'
};
exports.default = _default;

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement(_.Spinner, null);
};

exports.Default = Default;

var Multiple = function Multiple() {
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_.Spinner, {
    size: 30,
    color: "red"
  }), /*#__PURE__*/_react.default.createElement(_.Spinner, null), /*#__PURE__*/_react.default.createElement(_.Spinner, {
    size: 50,
    color: "green"
  }), /*#__PURE__*/_react.default.createElement(_.Spinner, {
    size: 60,
    color: "blue"
  }));
};

exports.Multiple = Multiple;

var Thickness = function Thickness() {
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_.Spinner, {
    width: 1
  }), /*#__PURE__*/_react.default.createElement(_.Spinner, {
    width: 2
  }), /*#__PURE__*/_react.default.createElement(_.Spinner, null), /*#__PURE__*/_react.default.createElement(_.Spinner, {
    width: 4
  }));
};

exports.Thickness = Thickness;
