// vcc-ui/lib/components/spacer/stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MultipleSizes = exports.FlexDirection = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _flex = require("../flex");

var _button = require("../button");

var _ = require("./");

var _default = {
  title: 'Spacer'
};
exports.default = _default;

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement(_flex.Flex, null, /*#__PURE__*/_react.default.createElement(_button.Button, null, "Hello"), /*#__PURE__*/_react.default.createElement(_.Spacer, null), /*#__PURE__*/_react.default.createElement(_button.Button, null, "World"));
};

exports.Default = Default;

var FlexDirection = function FlexDirection() {
  return /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: {
      flexDirection: 'row'
    }
  }, /*#__PURE__*/_react.default.createElement(_button.Button, null, "Hello"), /*#__PURE__*/_react.default.createElement(_.Spacer, null), /*#__PURE__*/_react.default.createElement(_button.Button, null, "World"));
};

exports.FlexDirection = FlexDirection;

var MultipleSizes = function MultipleSizes() {
  return /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: {
      flexDirection: 'row'
    }
  }, /*#__PURE__*/_react.default.createElement(_button.Button, null, "First"), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: {
      default: 2,
      fromL: 8
    }
  }), /*#__PURE__*/_react.default.createElement(_button.Button, null, "Second"), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: [2, 2, 8]
  }), /*#__PURE__*/_react.default.createElement(_button.Button, null, "Third"));
};

exports.MultipleSizes = MultipleSizes;
