// vcc-ui/lib/components/row/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Row = Row;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactFela = require("react-fela");

var _flex = require("../flex");

var styles = {
  row: function row(_ref) {
    var theme = _ref.theme,
        align = _ref.align;
    return {
      boxSizing: 'border-box',
      justifyContent: align === 'center' ? 'center' : 'flex-' + align,
      marginLeft: -theme.baselineGrid,
      marginRight: -theme.baselineGrid,
      flexDirection: 'row',
      flexWrap: 'wrap',
      fromL: {
        marginLeft: theme.baselineGrid * -1.5,
        marginRight: theme.baselineGrid * -1.5
      }
    };
  }
};

function Row(_ref2) {
  var children = _ref2.children,
      _ref2$align = _ref2.align,
      align = _ref2$align === void 0 ? 'start' : _ref2$align;

  var _useFela = (0, _reactFela.useFela)(),
      theme = _useFela.theme;

  var styleProps = {
    theme: theme,
    align: align
  };
  return /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: styles.row(styleProps)
  }, children);
}

Row.propTypes = {
  /** Align columns */
  align: _propTypes.default.oneOf(['start', 'end', 'center']),

  /** A JSX node */
  children: _propTypes.default.node
};
