// vcc-ui/es/components/nav/stories.js
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import React from 'react';
import { loremIpsum } from '../../fixtures';
import { Nav } from './';
import { Flex } from '../flex';
import { NavItem } from '../nav-item';
import { Block } from '../block';
import { Logo } from '../logo';
export default {
  title: 'Nav'
};
export var Default = function Default() {
  return /*#__PURE__*/React.createElement(Block, null, /*#__PURE__*/React.createElement(Nav, null, /*#__PURE__*/React.createElement(Flex, {
    extend: function extend(_ref) {
      var theme = _ref.theme;
      return _defineProperty({}, theme.breakpoints.fromL, {
        flexDirection: 'row'
      });
    }
  }, /*#__PURE__*/React.createElement(NavItem, null, "This is Volvo"), /*#__PURE__*/React.createElement(NavItem, {
    isActive: true
  }, "Build & Price"), /*#__PURE__*/React.createElement(NavItem, {
    isDropdown: true
  }, "Cars"), /*#__PURE__*/React.createElement(NavItem, {
    isDropdown: true,
    isActive: true
  }, "Bicycles"))), /*#__PURE__*/React.createElement(Block, {
    extend: {
      marginTop: '20px',
      fontSize: '48px',
      width: '320px'
    }
  }, loremIpsum));
};
export var Sticky = function Sticky() {
  return /*#__PURE__*/React.createElement(Block, null, /*#__PURE__*/React.createElement(Nav, {
    sticky: true
  }, /*#__PURE__*/React.createElement(Block, {
    extend: {
      padding: '20px'
    }
  }, /*#__PURE__*/React.createElement(Logo, {
    type: "square"
  }))), /*#__PURE__*/React.createElement(Block, {
    extend: {
      marginTop: '100px',
      fontSize: '48px',
      width: '320px'
    }
  }, loremIpsum));
};
export var HideOnScroll = function HideOnScroll() {
  return /*#__PURE__*/React.createElement(Block, null, /*#__PURE__*/React.createElement(Nav, {
    sticky: true,
    hideOnScroll: true
  }, /*#__PURE__*/React.createElement(Block, {
    extend: {
      padding: '20px'
    }
  }, /*#__PURE__*/React.createElement(Logo, {
    type: "square"
  }))), /*#__PURE__*/React.createElement(Block, {
    extend: {
      marginTop: '100px',
      fontSize: '48px',
      width: '320px'
    }
  }, loremIpsum));
};
