// vcc-ui/lib/components/view/stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Spacing = exports.Responsive = exports.Row = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = require("./");

var _default = {
  title: 'View'
};
exports.default = _default;
var style = {
  background: 'grey',
  padding: 20,
  margin: 5
};

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement(_.View, null, /*#__PURE__*/_react.default.createElement(_.View, {
    extend: style
  }, "Hello"), /*#__PURE__*/_react.default.createElement(_.View, {
    extend: style
  }, "World"));
};

exports.Default = Default;

var Row = function Row() {
  return /*#__PURE__*/_react.default.createElement(_.View, {
    direction: "row"
  }, /*#__PURE__*/_react.default.createElement(_.View, {
    extend: style
  }, "Hello"), /*#__PURE__*/_react.default.createElement(_.View, {
    extend: style
  }, "World"));
};

exports.Row = Row;

var Responsive = function Responsive() {
  return /*#__PURE__*/_react.default.createElement(_.View, {
    direction: ['column', 'row']
  }, /*#__PURE__*/_react.default.createElement(_.View, {
    extend: style
  }, "Column on Mobile"), /*#__PURE__*/_react.default.createElement(_.View, {
    extend: style
  }, "Row on Desktop (fromL)"));
};

exports.Responsive = Responsive;

var Spacing = function Spacing() {
  return /*#__PURE__*/_react.default.createElement(_.View, {
    spacing: 2
  }, "Spacing: 0", /*#__PURE__*/_react.default.createElement(_.View, {
    extend: style
  }, /*#__PURE__*/_react.default.createElement(_.View, {
    padding: 2,
    extend: {
      background: 'lightgrey'
    }
  }), /*#__PURE__*/_react.default.createElement(_.View, {
    padding: 2,
    extend: {
      background: 'lightgrey'
    }
  })), "Spacing: 1", /*#__PURE__*/_react.default.createElement(_.View, {
    extend: style,
    spacing: 1
  }, /*#__PURE__*/_react.default.createElement(_.View, {
    padding: 2,
    extend: {
      background: 'lightgrey'
    }
  }), /*#__PURE__*/_react.default.createElement(_.View, {
    padding: 2,
    extend: {
      background: 'lightgrey'
    }
  })), "Spacing: 2", /*#__PURE__*/_react.default.createElement(_.View, {
    extend: style,
    spacing: 2
  }, /*#__PURE__*/_react.default.createElement(_.View, {
    padding: 2,
    extend: {
      background: 'lightgrey'
    }
  }), /*#__PURE__*/_react.default.createElement(_.View, {
    padding: 2,
    extend: {
      background: 'lightgrey'
    }
  })));
};

exports.Spacing = Spacing;
