// vcc-ui/es/components/radio/stories.js
import React from 'react';
import { Radio } from '.';
import { Flex } from '../flex';
import { Text } from '../text';
import { Spacer } from '../spacer';
import { View } from '../view';
export default {
  title: 'Radio'
};

var DisplayWrapper = function DisplayWrapper(_ref) {
  var name = _ref.name,
      children = _ref.children;
  return /*#__PURE__*/React.createElement(Flex, {
    extend: {
      padding: 20,
      paddingRight: 40,
      minWidth: 200
    }
  }, /*#__PURE__*/React.createElement(Text, null, name), /*#__PURE__*/React.createElement(Spacer, {
    size: 0.5
  }), children);
};

export var Default = function Default() {
  function RadioWithValue(_ref2) {
    var _ref2$defaultValue = _ref2.defaultValue,
        defaultValue = _ref2$defaultValue === void 0 ? false : _ref2$defaultValue,
        disabled = _ref2.disabled,
        isValid = _ref2.isValid;
    return /*#__PURE__*/React.createElement(Radio, {
      disabled: disabled,
      onChange: function onChange() {},
      label: "Radio",
      checked: defaultValue,
      value: defaultValue,
      isValid: isValid
    });
  }

  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Flex, {
    extend: {
      flexDirection: 'row',
      paddingBottom: 20
    }
  }, /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Unchecked"
  }, /*#__PURE__*/React.createElement(RadioWithValue, null)), /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Checked"
  }, /*#__PURE__*/React.createElement(RadioWithValue, {
    defaultValue: true
  })), /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Error"
  }, /*#__PURE__*/React.createElement(RadioWithValue, {
    isValid: false
  }))), /*#__PURE__*/React.createElement(Flex, {
    extend: {
      flexDirection: 'row',
      paddingBottom: 20
    }
  }, /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Unchecked - Disabled"
  }, /*#__PURE__*/React.createElement(RadioWithValue, {
    disabled: true
  })), /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Checked - Disabled"
  }, /*#__PURE__*/React.createElement(RadioWithValue, {
    disabled: true,
    defaultValue: true
  }))));
};
export var Container = function Container() {
  return /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Checkbox within a direction=row container"
  }, /*#__PURE__*/React.createElement(View, {
    direction: "row",
    width: 200,
    bg: "rgb(240, 240, 240)",
    padding: 1
  }, /*#__PURE__*/React.createElement(Radio, {
    onChange: function onChange() {},
    label: "I am a super long label and I should wrap within my container",
    checked: false,
    value: false
  })));
};
