// vcc-ui/lib/components/grid/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Grid = Grid;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _flex = require("../flex");

var styles = {
  grid: function grid(_ref) {
    var theme = _ref.theme;
    return {
      boxSizing: 'border-box',
      width: '100%',
      maxWidth: 1400,
      margin: '0 auto',
      paddingLeft: theme.baselineGrid * 2,
      paddingRight: theme.baselineGrid * 2,
      // We use onlyM here so that breakpoint orders don't need to be defined by the user
      onlyM: {
        paddingLeft: theme.baselineGrid * 3,
        paddingRight: theme.baselineGrid * 3
      },
      fromL: {
        paddingLeft: theme.baselineGrid * 5,
        paddingRight: theme.baselineGrid * 5
      }
    };
  }
};

function Grid(_ref2) {
  var children = _ref2.children;
  return /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: styles.grid
  }, children);
}

Grid.propTypes = {
  /** A JSX node */
  children: _propTypes.default.node
};
