// vcc-ui/lib/components/grid/stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Responsive = exports.NestedCols = exports.Default = exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _ = require("./");

var _row = require("../row");

var _col = require("../col");

var _debugGrid = require("../debug-grid");

var _default = {
  title: 'Grid'
};
exports.default = _default;

var DebugBorder = function DebugBorder(props) {
  return /*#__PURE__*/_react.default.createElement("div", (0, _extends2.default)({
    style: {
      border: '1px solid grey',
      minHeight: '40px',
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      margin: '5px 0'
    }
  }, props));
};

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_debugGrid.DebugGrid, {
    storybook: true
  }), /*#__PURE__*/_react.default.createElement(_.Grid, null, /*#__PURE__*/_react.default.createElement("h2", null, "Center"), /*#__PURE__*/_react.default.createElement(_row.Row, {
    align: "center"
  }, /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 2
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      backgroundColor: 'red'
    }
  }, "size=2")), /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 1
  }, /*#__PURE__*/_react.default.createElement(DebugBorder, null, "size=1 (default)")), /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 7
  }, /*#__PURE__*/_react.default.createElement(DebugBorder, null, "size=7"))), /*#__PURE__*/_react.default.createElement(_row.Row, {
    align: "center"
  }, /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 2
  }, /*#__PURE__*/_react.default.createElement(DebugBorder, null, "size=2")), /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 1
  }, /*#__PURE__*/_react.default.createElement(DebugBorder, null, "size=1 (default)")), /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 6
  }, /*#__PURE__*/_react.default.createElement(DebugBorder, null, "size=6")), /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 1
  })), /*#__PURE__*/_react.default.createElement(_row.Row, {
    align: "center"
  }, /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 1
  }), /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 2
  }, /*#__PURE__*/_react.default.createElement(DebugBorder, null, "size=2")), /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 1
  }, /*#__PURE__*/_react.default.createElement(DebugBorder, null, "size=1 (default)")), /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 6
  }, /*#__PURE__*/_react.default.createElement(DebugBorder, null, "size=6"))), /*#__PURE__*/_react.default.createElement("h2", null, "Start"), /*#__PURE__*/_react.default.createElement(_row.Row, {
    align: "start"
  }, /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 2
  }, /*#__PURE__*/_react.default.createElement(DebugBorder, null, "size=2")), /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 1
  }, /*#__PURE__*/_react.default.createElement(DebugBorder, null, "size=1 (default)")), /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 7
  }, /*#__PURE__*/_react.default.createElement(DebugBorder, null, "size=7"))), /*#__PURE__*/_react.default.createElement("h2", null, "End"), /*#__PURE__*/_react.default.createElement(_row.Row, {
    align: "end"
  }, /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 2
  }, /*#__PURE__*/_react.default.createElement(DebugBorder, null, "size=2")), /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 1
  }, /*#__PURE__*/_react.default.createElement(DebugBorder, null, "size=1 (default)")), /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 7
  }, /*#__PURE__*/_react.default.createElement(DebugBorder, null, "size=7")))));
};

exports.Default = Default;

var NestedCols = function NestedCols() {
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_debugGrid.DebugGrid, {
    storybook: true
  }), /*#__PURE__*/_react.default.createElement(_.Grid, null, /*#__PURE__*/_react.default.createElement(_row.Row, null, /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 3
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      backgroundColor: 'grey'
    }
  }, "size=3")), /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 7
  }, /*#__PURE__*/_react.default.createElement(_row.Row, null, /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 3
  }, /*#__PURE__*/_react.default.createElement(_row.Row, null, /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 1
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      backgroundColor: 'blue'
    }
  }, "size=1")), /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 2
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      backgroundColor: 'green'
    }
  }, "size=2")))), /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 4
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      backgroundColor: 'green'
    }
  }, "size=4"))), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement(_row.Row, null, /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 5
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      backgroundColor: 'blue'
    }
  }, "size=5")), /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 1
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      backgroundColor: 'green'
    }
  }, "size=1")))))));
};

exports.NestedCols = NestedCols;

var Responsive = function Responsive() {
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_debugGrid.DebugGrid, {
    storybook: true
  }), /*#__PURE__*/_react.default.createElement(_.Grid, null, /*#__PURE__*/_react.default.createElement(_row.Row, null, /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 1
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      backgroundColor: 'purple'
    }
  }, "Hello world"))), /*#__PURE__*/_react.default.createElement(_row.Row, null, /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 2
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      backgroundColor: 'blue'
    }
  }, "Hello world")), /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 2
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      backgroundColor: 'blue'
    }
  }, "Hello world")), /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 2
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      backgroundColor: 'blue'
    }
  }, "Hello world")), /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: 2
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      backgroundColor: 'blue'
    }
  }, "Hello world"))), /*#__PURE__*/_react.default.createElement(_row.Row, null, /*#__PURE__*/_react.default.createElement(_col.Col, null, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      backgroundColor: 'green'
    }
  }, "I am full width always"))), /*#__PURE__*/_react.default.createElement(_row.Row, {
    align: "center"
  }, /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: {
      default: 4,
      '@media (min-width: 480px)': 3,
      '@media (min-width: 1024px)': 6,
      '@media (min-width: 1600px)': 10
    }
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      backgroundColor: 'red'
    }
  }, "My size responds to your viewport width"))), /*#__PURE__*/_react.default.createElement(_row.Row, {
    align: "center"
  }, /*#__PURE__*/_react.default.createElement(_col.Col, {
    size: {
      default: 4,
      fromM: 3,
      fromL: 6,
      fromXL: 10
    }
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      backgroundColor: 'red'
    }
  }, "My size responds to your viewport width using named keys")))));
};

exports.Responsive = Responsive;
