// vcc-ui/es/components/button/stories.js
import React from 'react';
import { Button } from './';
import { Text } from '../text';
import { View } from '../view';
export default {
  title: 'Button'
};

var DisplayWrapper = function DisplayWrapper(_ref) {
  var name = _ref.name,
      children = _ref.children;
  return /*#__PURE__*/React.createElement(View, {
    spacing: 1,
    extend: {
      minWidth: 250
    }
  }, /*#__PURE__*/React.createElement(Text, null, name), children);
};

var Row = function Row(_ref2) {
  var children = _ref2.children;
  return /*#__PURE__*/React.createElement(View, {
    direction: ['column', 'row'],
    spacing: 4
  }, children);
};

export var Default = function Default() {
  return /*#__PURE__*/React.createElement(View, {
    spacing: 4
  }, /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Primary - Default"
  }, /*#__PURE__*/React.createElement(Button, null, "Button")), /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Primary - Disabled"
  }, /*#__PURE__*/React.createElement(Button, {
    disabled: true
  }, "Button")), /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Primary - Loading"
  }, /*#__PURE__*/React.createElement(Button, {
    loading: true
  }, "Button"))), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Primary/Outline - Default"
  }, /*#__PURE__*/React.createElement(Button, {
    variant: "outline"
  }, "Button")), /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Primary/Outline - Disabled"
  }, /*#__PURE__*/React.createElement(Button, {
    variant: "outline",
    disabled: true
  }, "Button")), /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Primary/Outline - Loading"
  }, /*#__PURE__*/React.createElement(Button, {
    variant: "outline",
    loading: true
  }, "Button"))), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Primary/Text - Default"
  }, /*#__PURE__*/React.createElement(Button, {
    variant: "text"
  }, "Button")), /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Primary/Text - Disabled"
  }, /*#__PURE__*/React.createElement(Button, {
    variant: "text",
    disabled: true
  }, "Button")), /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Primary/Text - Loading"
  }, /*#__PURE__*/React.createElement(Button, {
    variant: "text",
    loading: true
  }, "Button"))), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Secondary - Default"
  }, /*#__PURE__*/React.createElement(Button, {
    intent: "secondary"
  }, "Button")), /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Secondary - Disabled"
  }, /*#__PURE__*/React.createElement(Button, {
    intent: "secondary",
    disabled: true
  }, "Button")), /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Secondary - Loading"
  }, /*#__PURE__*/React.createElement(Button, {
    intent: "secondary",
    loading: true
  }, "Button"))), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Secondary/Outline - Default"
  }, /*#__PURE__*/React.createElement(Button, {
    intent: "secondary",
    variant: "outline"
  }, "Button")), /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Secondary/Outline - Disabled"
  }, /*#__PURE__*/React.createElement(Button, {
    intent: "secondary",
    variant: "outline",
    disabled: true
  }, "Button")), /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Secondary/Outline - Loading"
  }, /*#__PURE__*/React.createElement(Button, {
    intent: "secondary",
    variant: "outline",
    loading: true
  }, "Button"))), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Secondary/Text - Default"
  }, /*#__PURE__*/React.createElement(Button, {
    intent: "secondary",
    variant: "text"
  }, "Button")), /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Secondary/Text - Disabled"
  }, /*#__PURE__*/React.createElement(Button, {
    intent: "secondary",
    variant: "text",
    disabled: true
  }, "Button")), /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Secondary/Text - Loading"
  }, /*#__PURE__*/React.createElement(Button, {
    intent: "secondary",
    variant: "text",
    loading: true
  }, "Button"))), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Destructive - Default"
  }, /*#__PURE__*/React.createElement(Button, {
    intent: "destructive"
  }, "Button")), /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Destructive - Disabled"
  }, /*#__PURE__*/React.createElement(Button, {
    intent: "destructive",
    disabled: true
  }, "Button")), /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Destructive - Loading"
  }, /*#__PURE__*/React.createElement(Button, {
    intent: "destructive",
    loading: true
  }, "Button"))), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Destructive/Outline - Default"
  }, /*#__PURE__*/React.createElement(Button, {
    intent: "destructive",
    variant: "outline"
  }, "Button")), /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Destructive/Outline - Disabled"
  }, /*#__PURE__*/React.createElement(Button, {
    intent: "destructive",
    variant: "outline",
    disabled: true
  }, "Button")), /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Destructive/Outline - Loading"
  }, /*#__PURE__*/React.createElement(Button, {
    intent: "destructive",
    variant: "outline",
    loading: true
  }, "Button"))), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Destructive/Text - Default"
  }, /*#__PURE__*/React.createElement(Button, {
    intent: "destructive",
    variant: "text"
  }, "Button")), /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Destructive/Outline - Disabled"
  }, /*#__PURE__*/React.createElement(Button, {
    intent: "destructive",
    variant: "text",
    disabled: true
  }, "Button")), /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Destructive/Outline - Loading"
  }, /*#__PURE__*/React.createElement(Button, {
    intent: "destructive",
    variant: "text",
    loading: true
  }, "Button"))));
};
export var Sizes = function Sizes() {
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Small"
  }, /*#__PURE__*/React.createElement(Button, {
    size: "small"
  }, "Button")), /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Medium"
  }, /*#__PURE__*/React.createElement(Button, {
    size: "medium"
  }, "Button")), /*#__PURE__*/React.createElement(DisplayWrapper, {
    name: "Large"
  }, /*#__PURE__*/React.createElement(Button, {
    size: "large"
  }, "Button"))));
};
