// vcc-ui/lib/components/radio/stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Container = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = require(".");

var _flex = require("../flex");

var _text = require("../text");

var _spacer = require("../spacer");

var _view = require("../view");

var _default = {
  title: 'Radio'
};
exports.default = _default;

var DisplayWrapper = function DisplayWrapper(_ref) {
  var name = _ref.name,
      children = _ref.children;
  return /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: {
      padding: 20,
      paddingRight: 40,
      minWidth: 200
    }
  }, /*#__PURE__*/_react.default.createElement(_text.Text, null, name), /*#__PURE__*/_react.default.createElement(_spacer.Spacer, {
    size: 0.5
  }), children);
};

var Default = function Default() {
  function RadioWithValue(_ref2) {
    var _ref2$defaultValue = _ref2.defaultValue,
        defaultValue = _ref2$defaultValue === void 0 ? false : _ref2$defaultValue,
        disabled = _ref2.disabled,
        isValid = _ref2.isValid;
    return /*#__PURE__*/_react.default.createElement(_.Radio, {
      disabled: disabled,
      onChange: function onChange() {},
      label: "Radio",
      checked: defaultValue,
      value: defaultValue,
      isValid: isValid
    });
  }

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: {
      flexDirection: 'row',
      paddingBottom: 20
    }
  }, /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Unchecked"
  }, /*#__PURE__*/_react.default.createElement(RadioWithValue, null)), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Checked"
  }, /*#__PURE__*/_react.default.createElement(RadioWithValue, {
    defaultValue: true
  })), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Error"
  }, /*#__PURE__*/_react.default.createElement(RadioWithValue, {
    isValid: false
  }))), /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: {
      flexDirection: 'row',
      paddingBottom: 20
    }
  }, /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Unchecked - Disabled"
  }, /*#__PURE__*/_react.default.createElement(RadioWithValue, {
    disabled: true
  })), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Checked - Disabled"
  }, /*#__PURE__*/_react.default.createElement(RadioWithValue, {
    disabled: true,
    defaultValue: true
  }))));
};

exports.Default = Default;

var Container = function Container() {
  return /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Checkbox within a direction=row container"
  }, /*#__PURE__*/_react.default.createElement(_view.View, {
    direction: "row",
    width: 200,
    bg: "rgb(240, 240, 240)",
    padding: 1
  }, /*#__PURE__*/_react.default.createElement(_.Radio, {
    onChange: function onChange() {},
    label: "I am a super long label and I should wrap within my container",
    checked: false,
    value: false
  })));
};

exports.Container = Container;
