// vcc-ui/es/components/view/index.js
import _extends from "@babel/runtime/helpers/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
import PropTypes from 'prop-types';
import React, { Children, forwardRef } from 'react';
import { useFela } from 'react-fela';
import { Spacer } from '../spacer';
import { applySpaceMultiplier } from '../../apply-space-multiplier';
import { responsiveProp, ruleType, responsiveNumberProp, responsiveStringProp } from '../../prop-types';
export var View = /*#__PURE__*/forwardRef(function (_ref, ref) {
  var children = _ref.children,
      As = _ref.as,
      extend = _ref.extend,
      inlineStyle = _ref.style,
      space = _ref.spacing,
      className = _ref.className,
      bg = _ref.bg,
      _ref$backgroundColor = _ref.backgroundColor,
      backgroundColor = _ref$backgroundColor === void 0 ? bg : _ref$backgroundColor,
      padding = _ref.padding,
      paddingLeft = _ref.paddingLeft,
      paddingRight = _ref.paddingRight,
      paddingBottom = _ref.paddingBottom,
      paddingTop = _ref.paddingTop,
      margin = _ref.margin,
      marginLeft = _ref.marginLeft,
      marginRight = _ref.marginRight,
      marginBottom = _ref.marginBottom,
      marginTop = _ref.marginTop,
      height = _ref.height,
      width = _ref.width,
      grow = _ref.grow,
      shrink = _ref.shrink,
      minWidth = _ref.minWidth,
      maxWidth = _ref.maxWidth,
      minHeight = _ref.minHeight,
      maxHeight = _ref.maxHeight,
      order = _ref.order,
      alignContent = _ref.alignContent,
      justifyContent = _ref.justifyContent,
      alignItems = _ref.alignItems,
      alignSelf = _ref.alignSelf,
      flex = _ref.flex,
      basis = _ref.basis,
      direction = _ref.direction,
      display = _ref.display,
      wrap = _ref.wrap,
      props = _objectWithoutProperties(_ref, ["children", "as", "extend", "style", "spacing", "className", "bg", "backgroundColor", "padding", "paddingLeft", "paddingRight", "paddingBottom", "paddingTop", "margin", "marginLeft", "marginRight", "marginBottom", "marginTop", "height", "width", "grow", "shrink", "minWidth", "maxWidth", "minHeight", "maxHeight", "order", "alignContent", "justifyContent", "alignItems", "alignSelf", "flex", "basis", "direction", "display", "wrap"]);

  var _useFela = useFela(props),
      css = _useFela.css,
      theme = _useFela.theme;

  var spacing = applySpaceMultiplier(theme.baselineGrid);
  return /*#__PURE__*/React.createElement(As, _extends({}, props, {
    style: inlineStyle,
    ref: ref,
    className: css({
      boxSizing: 'border-box',
      flexDirection: direction,
      flexWrap: wrap,
      flexGrow: grow,
      flexShrink: shrink,
      flexBasis: basis,
      flex: flex,
      justifyContent: justifyContent,
      alignContent: alignContent,
      alignItems: alignItems,
      alignSelf: alignSelf,
      order: order,
      display: display,
      maxWidth: maxWidth,
      minWidth: minWidth,
      width: width,
      maxHeight: maxHeight,
      minHeight: minHeight,
      height: height,
      padding: spacing(padding),
      paddingLeft: spacing(paddingLeft),
      paddingRight: spacing(paddingRight),
      paddingBottom: spacing(paddingBottom),
      paddingTop: spacing(paddingTop),
      margin: spacing(margin),
      marginLeft: spacing(marginLeft),
      marginRight: spacing(marginRight),
      marginBottom: spacing(marginBottom),
      marginTop: spacing(marginTop),
      backgroundColor: backgroundColor
    }, extend) + (className ? ' ' + className : '')
  }), space ? Children.toArray(children).map(function (child, index, arr) {
    return /*#__PURE__*/React.createElement(React.Fragment, {
      key: index
    }, child, index === arr.length - 1 ? null : /*#__PURE__*/React.createElement(Spacer, {
      size: space
    }));
  }) : children);
});
View.displayName = 'View';
View.defaultProps = {
  direction: 'column',
  as: 'div',
  grow: 0,
  shrink: 0,
  basis: 'auto',
  display: 'flex',
  alignItems: 'stretch',
  wrap: 'nowrap'
};
var directionProp = PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.oneOf(['row', 'column', 'row-reverse', 'column-reverse'])), PropTypes.oneOf(['row', 'column', 'row-reverse', 'column-reverse'])]);
View.propTypes = {
  /** A JSX node */
  children: PropTypes.node,

  /** The HTML node that is rendered. */
  as: PropTypes.oneOfType([PropTypes.string, PropTypes.element, PropTypes.elementType, PropTypes.func]),

  /** Adds a custom CSS class. */
  className: PropTypes.string,

  /** Extends the Fela style object. */
  extend: PropTypes.oneOfType([ruleType, PropTypes.arrayOf(ruleType)]),

  /** Adds inline styles. */
  style: PropTypes.object,

  /** A custom background color */
  bg: responsiveStringProp,

  /** A custom background color */
  backgroundColor: responsiveStringProp,

  /** Adds spacing between children based on the baselineGrid. */
  spacing: PropTypes.oneOfType([responsiveNumberProp, responsiveStringProp]),

  /** Adds left padding based on the baselineGrid. */
  paddingLeft: responsiveNumberProp,

  /** Adds right padding based on the baselineGrid. */
  paddingRight: responsiveProp,

  /** Adds bottom padding based on the baselineGrid. */
  paddingBottom: responsiveNumberProp,

  /** Adds top padding based on the baselineGrid. */
  paddingTop: responsiveNumberProp,

  /** Adds padding based on the baselineGrid. Overwritten by specific directional paddings. */
  padding: responsiveNumberProp,

  /** Adds left margin based on the baselineGrid. */
  marginLeft: responsiveNumberProp,

  /** Adds right margin based on the baselineGrid. */
  marginRight: responsiveNumberProp,

  /** Adds bottom margin based on the baselineGrid. */
  marginBottom: responsiveNumberProp,

  /** Adds top margin based on the baselineGrid. */
  marginTop: responsiveNumberProp,

  /** Adds margin based on the baselineGrid. Overwritten by specific directional margins. */
  margin: responsiveNumberProp,

  /** Sets display. */
  display: responsiveStringProp,

  /** Sets flex-wrap. */
  wrap: responsiveStringProp,

  /** Sets the flex-direction. */
  direction: directionProp,

  /** Sets flex-grow. */
  grow: responsiveNumberProp,

  /** Sets flex-shrink. */
  shrink: responsiveNumberProp,

  /** Sets flex-basis. */
  basis: responsiveProp,

  /** Sets order. */
  order: responsiveNumberProp,

  /** Sets flex. */
  flex: responsiveProp,

  /** Sets justify-content. */
  justifyContent: responsiveStringProp,

  /** Sets align-content. */
  alignContent: responsiveStringProp,

  /** Sets align-items. */
  alignItems: responsiveStringProp,

  /** Sets align-self. */
  alignSelf: responsiveStringProp,

  /** Sets max-width. */
  maxWidth: responsiveProp,

  /** Sets min-width. */
  minWidth: responsiveProp,

  /** Sets width. */
  width: responsiveProp,

  /** Sets max-height. */
  maxHeight: responsiveProp,

  /** Sets min-height. */
  minHeight: responsiveProp,

  /** Sets height. */
  height: responsiveProp
};
