// vcc-ui/es/components/view/stories.js
import React from 'react';
import { View } from './';
export default {
  title: 'View'
};
var style = {
  background: 'grey',
  padding: 20,
  margin: 5
};
export var Default = function Default() {
  return /*#__PURE__*/React.createElement(View, null, /*#__PURE__*/React.createElement(View, {
    extend: style
  }, "Hello"), /*#__PURE__*/React.createElement(View, {
    extend: style
  }, "World"));
};
export var Row = function Row() {
  return /*#__PURE__*/React.createElement(View, {
    direction: "row"
  }, /*#__PURE__*/React.createElement(View, {
    extend: style
  }, "Hello"), /*#__PURE__*/React.createElement(View, {
    extend: style
  }, "World"));
};
export var Responsive = function Responsive() {
  return /*#__PURE__*/React.createElement(View, {
    direction: ['column', 'row']
  }, /*#__PURE__*/React.createElement(View, {
    extend: style
  }, "Column on Mobile"), /*#__PURE__*/React.createElement(View, {
    extend: style
  }, "Row on Desktop (fromL)"));
};
export var Spacing = function Spacing() {
  return /*#__PURE__*/React.createElement(View, {
    spacing: 2
  }, "Spacing: 0", /*#__PURE__*/React.createElement(View, {
    extend: style
  }, /*#__PURE__*/React.createElement(View, {
    padding: 2,
    extend: {
      background: 'lightgrey'
    }
  }), /*#__PURE__*/React.createElement(View, {
    padding: 2,
    extend: {
      background: 'lightgrey'
    }
  })), "Spacing: 1", /*#__PURE__*/React.createElement(View, {
    extend: style,
    spacing: 1
  }, /*#__PURE__*/React.createElement(View, {
    padding: 2,
    extend: {
      background: 'lightgrey'
    }
  }), /*#__PURE__*/React.createElement(View, {
    padding: 2,
    extend: {
      background: 'lightgrey'
    }
  })), "Spacing: 2", /*#__PURE__*/React.createElement(View, {
    extend: style,
    spacing: 2
  }, /*#__PURE__*/React.createElement(View, {
    padding: 2,
    extend: {
      background: 'lightgrey'
    }
  }), /*#__PURE__*/React.createElement(View, {
    padding: 2,
    extend: {
      background: 'lightgrey'
    }
  })));
};
