// vcc-ui/lib/components/card/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Card = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactFela = require("react-fela");

var _flex = require("../flex");

var _getThemeStyle = require("../../get-theme-style");

var transition = '300ms cubic-bezier(0.23, 1, 0.32, 1) 0s';

var style = function style(_ref) {
  var theme = _ref.theme,
      href = _ref.href,
      selected = _ref.selected;
  return {
    boxSizing: 'border-box',
    boxShadow: theme.tokens.card.shadow,
    overflow: 'hidden',
    background: theme.tokens.card.background,
    borderRadius: 4,
    extend: [{
      condition: selected,
      style: {
        position: 'relative',
        ':after': {
          content: '" "',
          pointerEvents: 'none',
          display: 'block',
          position: 'absolute',
          top: 0,
          bottom: 0,
          left: 0,
          right: 0,
          borderRadius: 4,
          border: '1px solid ' + theme.color.ornament.highlight
        }
      }
    }, {
      condition: href,
      style: {
        textDecoration: 'none',
        color: 'inherit',
        transition: "transform ".concat(transition, ", box-shadow ").concat(transition),
        ':hover': {
          transform: 'translateY(-5px)',
          boxShadow: 'rgba(0, 0, 0, 0.2) 0px 12px 24px'
        }
      }
    }]
  };
};

var Card = /*#__PURE__*/_react.default.forwardRef(function (_ref2, ref) {
  var selected = _ref2.selected,
      href = _ref2.href,
      children = _ref2.children,
      props = (0, _objectWithoutProperties2.default)(_ref2, ["selected", "href", "children"]);

  var _useFela = (0, _reactFela.useFela)(),
      theme = _useFela.theme;

  var styleProps = {
    theme: theme,
    selected: selected,
    href: href
  };
  return /*#__PURE__*/_react.default.createElement(_flex.Flex, (0, _extends2.default)({
    as: href ? 'a' : 'div',
    ref: ref,
    href: href
  }, props, {
    extend: [style(styleProps), (0, _getThemeStyle.getThemeStyle)('card', theme, styleProps)]
  }), children);
});

exports.Card = Card;
Card.displayName = 'Card';
Card.propTypes = {
  selected: _propTypes.default.bool,
  href: _propTypes.default.string,
  children: _propTypes.default.node
};
