// vcc-ui/lib/components/card/stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.withSelectedImageAndHref = exports.withImage = exports.withNestedOnClick = exports.withHref = exports.Selected = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = require("../../");

var _2 = require("./");

var _cardContent = require("../card-content");

var _default = {
  title: 'Card'
};
exports.default = _default;

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement(_2.Card, null, /*#__PURE__*/_react.default.createElement(_cardContent.CardContent, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "ootah"
  }, "Card Component"), /*#__PURE__*/_react.default.createElement(_.Spacer, null), /*#__PURE__*/_react.default.createElement(_.Text, null, "This is a card")));
};

exports.Default = Default;

var Selected = function Selected() {
  return /*#__PURE__*/_react.default.createElement(_2.Card, {
    selected: true
  }, /*#__PURE__*/_react.default.createElement(_cardContent.CardContent, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "ootah"
  }, "Card Component"), /*#__PURE__*/_react.default.createElement(_.Spacer, null), /*#__PURE__*/_react.default.createElement(_.Text, null, "This is a card is in a selected state")));
};

exports.Selected = Selected;

var withHref = function withHref() {
  return /*#__PURE__*/_react.default.createElement(_2.Card, {
    href: "#test"
  }, /*#__PURE__*/_react.default.createElement(_cardContent.CardContent, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "ootah"
  }, "Card Component"), /*#__PURE__*/_react.default.createElement(_.Spacer, null), /*#__PURE__*/_react.default.createElement(_.Text, null, "This is a card has a href prop and is a link")));
};

exports.withHref = withHref;

var withNestedOnClick = function withNestedOnClick() {
  return /*#__PURE__*/_react.default.createElement(_2.Card, {
    onClick: function onClick() {
      alert('button clicked');
    }
  }, /*#__PURE__*/_react.default.createElement(_cardContent.CardContent, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "ootah"
  }, "Card Component"), /*#__PURE__*/_react.default.createElement(_.Spacer, null), /*#__PURE__*/_react.default.createElement(_.Text, null, "This is a card has an onClick handler and contains buttons with links"), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, {
    extend: {
      flexDirection: 'row'
    }
  }, /*#__PURE__*/_react.default.createElement(_.Button, {
    onClick: function onClick(e) {
      e.stopPropagation();
    },
    href: "#buy"
  }, "Buy Now"), /*#__PURE__*/_react.default.createElement(_.Button, {
    onClick: function onClick(e) {
      e.stopPropagation();
    },
    href: "#learn-more",
    variant: "text"
  }, "Learn More"))));
};

exports.withNestedOnClick = withNestedOnClick;

var withImage = function withImage() {
  return /*#__PURE__*/_react.default.createElement(_2.Card, null, /*#__PURE__*/_react.default.createElement(_.Flex, {
    extend: {
      justifyContent: 'center',
      padding: 32,
      background: 'linear-gradient(pink, blue)'
    }
  }, /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "bates",
    extend: {
      color: '#fff'
    }
  }, "Fake Image")), /*#__PURE__*/_react.default.createElement(_cardContent.CardContent, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "ootah"
  }, "Card Component"), /*#__PURE__*/_react.default.createElement(_.Spacer, null), /*#__PURE__*/_react.default.createElement(_.Text, null, "This is a card component")));
};

exports.withImage = withImage;

var withSelectedImageAndHref = function withSelectedImageAndHref() {
  return /*#__PURE__*/_react.default.createElement(_2.Card, {
    selected: true,
    href: "#test"
  }, /*#__PURE__*/_react.default.createElement(_.Flex, {
    extend: {
      justifyContent: 'center',
      padding: 32,
      background: 'linear-gradient(pink, blue)'
    }
  }, /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "bates",
    extend: {
      color: '#fff'
    }
  }, "Fake Image")), /*#__PURE__*/_react.default.createElement(_cardContent.CardContent, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "ootah"
  }, "Card Component"), /*#__PURE__*/_react.default.createElement(_.Spacer, null), /*#__PURE__*/_react.default.createElement(_.Text, null, "This is a card component")));
};

exports.withSelectedImageAndHref = withSelectedImageAndHref;
