// vcc-ui/es/components/checkbox/index.js
import _extends from "@babel/runtime/helpers/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
import React from 'react';
import { bool, func, string, node, oneOfType } from 'prop-types';
import { useFela } from 'react-fela';
import { Block } from '../block';
import { View } from '../view';
import { Message } from '../message';
import { getThemeStyle } from '../../get-theme-style';
import { useId } from '../../auto-id';
import { deprecate } from '../../deprecate';

var checkmarkIcon = function checkmarkIcon(color) {
  return "url(\"data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='12' height='10' viewBox='0 0 12 10' fill='none' %3E %3Cpath stroke='".concat(encodeURIComponent(color), "' d='M1 6L4 9L11 1' stroke-width='1.5' stroke-linecap='round' stroke-linejoin='round' %3E %3C/path%3E %3C/svg%3E\")");
};

var checkboxStyle = function checkboxStyle(_ref) {
  var theme = _ref.theme,
      isValid = _ref.isValid;
  return {
    boxSizing: 'border-box',
    borderWidth: 1,
    borderStyle: 'solid',
    borderRadius: 4,
    appearance: 'none',
    width: 20,
    height: 20,
    margin: 0,
    padding: 0,
    flexShrink: 0,
    flexGrow: 0,
    outline: 0,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    position: 'relative',
    cursor: 'pointer',
    borderColor: theme.tokens.inputControlBorder,
    backgroundColor: theme.tokens.inputBackground,
    extend: [{
      condition: !isValid,
      style: {
        borderColor: theme.color.foreground.alert,
        borderWidth: 2
      }
    }, {
      condition: isValid,
      style: {
        ':focus': {
          borderColor: theme.tokens.inputBorderFocus
        }
      }
    }, {
      condition: isValid,
      style: {
        ':active': {
          color: 'pink'
        }
      }
    }],
    ':checked': {
      backgroundColor: theme.tokens.inputControlBackground,
      borderColor: theme.tokens.inputControlBackground,
      backgroundImage: checkmarkIcon(theme.tokens.inputControlForeground),
      backgroundSize: '12px 10px',
      backgroundPosition: 'center',
      backgroundRepeat: 'no-repeat'
    },
    ':disabled': {
      cursor: 'not-allowed',
      borderColor: theme.tokens.inputDisabledBorder,
      backgroundColor: theme.tokens.inputDisabledBackground,
      ':checked': {
        borderColor: theme.tokens.inputDisabledBorder,
        backgroundColor: theme.tokens.inputDisabledBackground,
        backgroundImage: checkmarkIcon(theme.tokens.inputDisabledControl)
      }
    }
  };
};

var labelStyle = function labelStyle(_ref2) {
  var theme = _ref2.theme,
      isValid = _ref2.isValid;
  return {
    color: isValid ? theme.color.foreground.secondary : theme.color.foreground.alert,
    fontSize: 16,
    letterSpacing: '0.02em',
    fontFamily: theme.fontTypes.NOVUM,
    fontWeight: 300
  };
};

export var Checkbox = /*#__PURE__*/React.forwardRef(function (_ref3, ref) {
  var _ref3$isValid = _ref3.isValid,
      isValid = _ref3$isValid === void 0 ? true : _ref3$isValid,
      _ref3$label = _ref3.label,
      label = _ref3$label === void 0 ? '' : _ref3$label,
      errorMessage = _ref3.errorMessage,
      description = _ref3.description,
      props = _objectWithoutProperties(_ref3, ["isValid", "label", "errorMessage", "description"]);

  var _useFela = useFela(),
      theme = _useFela.theme;

  var styleProps = {
    isValid: isValid,
    theme: theme
  };
  var id = useId('vcc-ui-checkbox', props.id); // legacy check

  var hasLabel = label && (typeof label === 'string' ? label.length > 0 : true);
  deprecate('Unlabelled text-input fields are deprecated and will be removed in 3.0.0. Pass a label prop.', !hasLabel);
  var messages = /*#__PURE__*/React.createElement(React.Fragment, null, !errorMessage ? null : /*#__PURE__*/React.createElement(Message, {
    type: "error"
  }, errorMessage), !description ? null : /*#__PURE__*/React.createElement(Message, null, description));
  return /*#__PURE__*/React.createElement(View, {
    spacing: 0.5,
    shrink: 1
  }, /*#__PURE__*/React.createElement(View, {
    spacing: 1,
    direction: "row"
  }, /*#__PURE__*/React.createElement(Block, _extends({}, props, {
    ref: ref,
    id: id,
    as: "input",
    type: "checkbox",
    extend: [checkboxStyle(styleProps), getThemeStyle('checkbox', theme, styleProps)]
  })), !hasLabel ? null : /*#__PURE__*/React.createElement(Block, {
    as: "label",
    htmlFor: id,
    style: labelStyle(styleProps)
  }, label)), messages);
});
Checkbox.displayName = 'Checkbox';
Checkbox.propTypes = {
  onChange: func.isRequired,
  checked: bool,
  isValid: bool,
  id: string,
  name: string,

  /** Renders a label inside the input.  */
  label: oneOfType([string, node]),

  /** Renders a description text underneath the input.  */
  description: string,

  /** Renders a red error message for validation underneath the input.  */
  errorMessage: string
};
