// vcc-ui/lib/components/slider/stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Disabled = exports.Fibonacci = exports.FiveSteps = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = require("./");

var _default = {
  title: 'Slider'
};
exports.default = _default;

var FiveSteps = function FiveSteps() {
  return /*#__PURE__*/_react.default.createElement(_.Slider, {
    onChange: function onChange(value) {
      // eslint-disable-next-line no-console
      console.log(value);
    },
    initialValue: 1,
    step: 1,
    minValue: 1,
    maxValue: 5
  });
};

exports.FiveSteps = FiveSteps;

var Fibonacci = function Fibonacci() {
  return /*#__PURE__*/_react.default.createElement(_.Slider, {
    initialValue: 1,
    onChange: function onChange(value) {
      // eslint-disable-next-line no-console
      console.log(value);
    },
    valueList: [0, 1, 2, 3, 5, 8, 13, 21, 34, 66, 89, 144]
  });
};

exports.Fibonacci = Fibonacci;

var Disabled = function Disabled() {
  return /*#__PURE__*/_react.default.createElement(_.Slider, {
    disabled: true,
    onChange: function onChange(value) {
      // eslint-disable-next-line no-console
      console.log(value);
    },
    initialValue: 3,
    step: 1,
    minValue: 1,
    maxValue: 5
  });
};

exports.Disabled = Disabled;
