// vcc-ui/lib/components/card-content/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CardContent = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactFela = require("react-fela");

var _flex = require("../flex");

var _getThemeStyle = require("../../get-theme-style");

var style = function style() {
  return {
    padding: '16px 32px'
  };
};

var CardContent = /*#__PURE__*/_react.default.forwardRef(function (_ref, ref) {
  var children = _ref.children,
      props = (0, _objectWithoutProperties2.default)(_ref, ["children"]);

  var _useFela = (0, _reactFela.useFela)(),
      theme = _useFela.theme;

  var styleProps = {
    theme: theme
  };
  return /*#__PURE__*/_react.default.createElement(_flex.Flex, (0, _extends2.default)({
    ref: ref
  }, props, {
    extend: [style(styleProps), (0, _getThemeStyle.getThemeStyle)('card-content', theme, styleProps)]
  }), children);
});

exports.CardContent = CardContent;
CardContent.displayName = 'CardContent';
CardContent.propTypes = {
  children: _propTypes.default.node
};
