// vcc-ui/es/components/spacer/stories.js
import React from 'react';
import { Flex } from '../flex';
import { Button } from '../button';
import { Spacer } from './';
export default {
  title: 'Spacer'
};
export var Default = function Default() {
  return /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Button, null, "Hello"), /*#__PURE__*/React.createElement(Spacer, null), /*#__PURE__*/React.createElement(Button, null, "World"));
};
export var FlexDirection = function FlexDirection() {
  return /*#__PURE__*/React.createElement(Flex, {
    extend: {
      flexDirection: 'row'
    }
  }, /*#__PURE__*/React.createElement(Button, null, "Hello"), /*#__PURE__*/React.createElement(Spacer, null), /*#__PURE__*/React.createElement(Button, null, "World"));
};
export var MultipleSizes = function MultipleSizes() {
  return /*#__PURE__*/React.createElement(Flex, {
    extend: {
      flexDirection: 'row'
    }
  }, /*#__PURE__*/React.createElement(Button, null, "First"), /*#__PURE__*/React.createElement(Spacer, {
    size: {
      default: 2,
      fromL: 8
    }
  }), /*#__PURE__*/React.createElement(Button, null, "Second"), /*#__PURE__*/React.createElement(Spacer, {
    size: [2, 2, 8]
  }), /*#__PURE__*/React.createElement(Button, null, "Third"));
};
