// vcc-ui/lib/components/text-input/stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Legacy = exports.Labelled = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = require("./");

var _flex = require("../flex");

var _text = require("../text");

var _spacer = require("../spacer");

var _default = {
  title: 'TextInput'
};
exports.default = _default;

var DisplayWrapper = function DisplayWrapper(_ref) {
  var name = _ref.name,
      children = _ref.children;
  return /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: {
      padding: 20,
      paddingRight: 40,
      minWidth: 300
    }
  }, /*#__PURE__*/_react.default.createElement(_text.Text, null, name), /*#__PURE__*/_react.default.createElement(_spacer.Spacer, {
    size: 0.5
  }), children);
};

var Labelled = function Labelled() {
  function TextInputWithValue(_ref2) {
    var _ref2$defaultValue = _ref2.defaultValue,
        defaultValue = _ref2$defaultValue === void 0 ? '' : _ref2$defaultValue,
        disabled = _ref2.disabled,
        loading = _ref2.loading,
        isValid = _ref2.isValid,
        errorMessage = _ref2.errorMessage,
        description = _ref2.description;
    return /*#__PURE__*/_react.default.createElement(_.TextInput, {
      disabled: disabled,
      label: "Label",
      name: "test",
      loading: loading,
      onChange: function onChange() {},
      description: description,
      errorMessage: errorMessage,
      placeholder: "e.g., Placeholder",
      value: defaultValue,
      isValid: isValid
    });
  }

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: {
      flexDirection: 'row',
      paddingBottom: 20
    }
  }, /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Empty"
  }, /*#__PURE__*/_react.default.createElement(TextInputWithValue, null)), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Filled"
  }, /*#__PURE__*/_react.default.createElement(TextInputWithValue, {
    defaultValue: "Entry"
  })), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Loading"
  }, /*#__PURE__*/_react.default.createElement(TextInputWithValue, {
    defaultValue: "Entry",
    loading: true
  }))), /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: {
      flexDirection: 'row',
      paddingBottom: 20
    }
  }, /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Empty - Error"
  }, /*#__PURE__*/_react.default.createElement(TextInputWithValue, {
    isValid: false,
    errorMessage: "This is an error.",
    description: "I'm a helper text."
  })), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Filled - Error"
  }, /*#__PURE__*/_react.default.createElement(TextInputWithValue, {
    defaultValue: "Entry",
    isValid: false,
    errorMessage: "This is an error."
  })), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Loading - Error"
  }, /*#__PURE__*/_react.default.createElement(TextInputWithValue, {
    defaultValue: "Entry",
    loading: true,
    isValid: false,
    errorMessage: "This is an error."
  }))), /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: {
      flexDirection: 'row',
      paddingBottom: 20
    }
  }, /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Empty - Disabled"
  }, /*#__PURE__*/_react.default.createElement(TextInputWithValue, {
    disabled: true,
    description: "I'm a helper text."
  })), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Filled - Disabled"
  }, /*#__PURE__*/_react.default.createElement(TextInputWithValue, {
    defaultValue: "Entry",
    disabled: true
  })), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Loading - Disabled"
  }, /*#__PURE__*/_react.default.createElement(TextInputWithValue, {
    defaultValue: "Entry",
    loading: true,
    disabled: true
  }))));
};

exports.Labelled = Labelled;

var Legacy = function Legacy() {
  function TextInputWithValue(_ref3) {
    var _ref3$defaultValue = _ref3.defaultValue,
        defaultValue = _ref3$defaultValue === void 0 ? '' : _ref3$defaultValue,
        disabled = _ref3.disabled,
        isValid = _ref3.isValid,
        errorMessage = _ref3.errorMessage,
        description = _ref3.description;
    return /*#__PURE__*/_react.default.createElement(_.TextInput, {
      disabled: disabled,
      errorMessage: errorMessage,
      description: description,
      onChange: function onChange() {},
      placeholder: "e.g., Placeholder",
      value: defaultValue,
      isValid: isValid
    });
  }

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: {
      flexDirection: 'row',
      paddingBottom: 20
    }
  }, /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Empty"
  }, /*#__PURE__*/_react.default.createElement(TextInputWithValue, null)), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Filled"
  }, /*#__PURE__*/_react.default.createElement(TextInputWithValue, {
    defaultValue: "Entry"
  }))), /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: {
      flexDirection: 'row',
      paddingBottom: 20
    }
  }, /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Empty - Error"
  }, /*#__PURE__*/_react.default.createElement(TextInputWithValue, {
    isValid: false,
    errorMessage: "with error text",
    description: "and helper text"
  })), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Filled - Error"
  }, /*#__PURE__*/_react.default.createElement(TextInputWithValue, {
    defaultValue: "Entry",
    isValid: false,
    errorMessage: "with error text"
  }))), /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: {
      flexDirection: 'row',
      paddingBottom: 20
    }
  }, /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Empty - Disabled"
  }, /*#__PURE__*/_react.default.createElement(TextInputWithValue, {
    disabled: true
  })), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Filled - Disabled"
  }, /*#__PURE__*/_react.default.createElement(TextInputWithValue, {
    defaultValue: "Entry",
    description: "with helper",
    disabled: true
  }))));
};

exports.Legacy = Legacy;
