// vcc-ui/lib/components/toggle/stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Disabled = exports.Invalid = exports.Default = exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _ = require("./");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

var _default = {
  title: 'Toggle'
};
exports.default = _default;

var Default = function Default() {
  var TogglingToggle = /*#__PURE__*/function (_Component) {
    (0, _inherits2.default)(TogglingToggle, _Component);

    var _super = _createSuper(TogglingToggle);

    function TogglingToggle() {
      var _this;

      (0, _classCallCheck2.default)(this, TogglingToggle);

      for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
        args[_key] = arguments[_key];
      }

      _this = _super.call.apply(_super, [this].concat(args));
      (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "state", {
        checked: false
      });
      return _this;
    }

    (0, _createClass2.default)(TogglingToggle, [{
      key: "render",
      value: function render() {
        var _this2 = this;

        return /*#__PURE__*/_react.default.createElement(_.Toggle, {
          checked: this.state.checked,
          onChange: function onChange() {
            return _this2.setState({
              checked: !_this2.state.checked
            });
          }
        });
      }
    }]);
    return TogglingToggle;
  }(_react.Component);

  return /*#__PURE__*/_react.default.createElement(TogglingToggle, null);
};

exports.Default = Default;

var Invalid = function Invalid() {
  return /*#__PURE__*/_react.default.createElement(_.Toggle, {
    onChange: function onChange() {},
    isValid: false
  });
};

exports.Invalid = Invalid;

var Disabled = function Disabled() {
  var TogglingToggle = /*#__PURE__*/function (_Component2) {
    (0, _inherits2.default)(TogglingToggle, _Component2);

    var _super2 = _createSuper(TogglingToggle);

    function TogglingToggle() {
      var _this3;

      (0, _classCallCheck2.default)(this, TogglingToggle);

      for (var _len2 = arguments.length, args = new Array(_len2), _key2 = 0; _key2 < _len2; _key2++) {
        args[_key2] = arguments[_key2];
      }

      _this3 = _super2.call.apply(_super2, [this].concat(args));
      (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this3), "state", {
        checked: false
      });
      return _this3;
    }

    (0, _createClass2.default)(TogglingToggle, [{
      key: "render",
      value: function render() {
        var _this4 = this;

        return /*#__PURE__*/_react.default.createElement(_.Toggle, {
          disabled: true,
          checked: this.state.checked,
          onChange: function onChange() {
            return _this4.setState({
              checked: !_this4.state.checked
            });
          }
        });
      }
    }]);
    return TogglingToggle;
  }(_react.Component);

  return /*#__PURE__*/_react.default.createElement(TogglingToggle, null);
};

exports.Disabled = Disabled;
