// vcc-ui/lib/components/select-input/stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.OptGroups = exports.Legacy = exports.Labelled = exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _ = require("./");

var _flex = require("../flex");

var _text = require("../text");

var _spacer = require("../spacer");

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

var _default = {
  title: 'SelectInput'
};
exports.default = _default;

var DisplayWrapper = function DisplayWrapper(_ref) {
  var name = _ref.name,
      children = _ref.children;
  return /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: {
      padding: 20,
      paddingRight: 40,
      minWidth: 300
    }
  }, /*#__PURE__*/_react.default.createElement(_text.Text, null, name), /*#__PURE__*/_react.default.createElement(_spacer.Spacer, {
    size: 0.5
  }), children);
};

var Labelled = function Labelled() {
  function SelectInputWithValue(_ref2) {
    var _ref2$defaultValue = _ref2.defaultValue,
        defaultValue = _ref2$defaultValue === void 0 ? '' : _ref2$defaultValue,
        helperText = _ref2.helperText,
        errorText = _ref2.errorText,
        disabled = _ref2.disabled,
        loading = _ref2.loading,
        isValid = _ref2.isValid;
    return /*#__PURE__*/_react.default.createElement(_.SelectInput, {
      disabled: disabled,
      loading: loading,
      name: "test",
      label: "Label",
      onChange: function onChange() {},
      errorMessage: errorText,
      description: helperText,
      value: defaultValue,
      isValid: isValid
    }, /*#__PURE__*/_react.default.createElement("option", {
      value: "entry"
    }, "Entry"), /*#__PURE__*/_react.default.createElement("option", {
      value: "c30",
      disabled: true
    }, "Volvo C30"), /*#__PURE__*/_react.default.createElement("option", {
      value: "c70"
    }, "Volvo C70"), /*#__PURE__*/_react.default.createElement("option", {
      value: "c90"
    }, "Volvo C90"), /*#__PURE__*/_react.default.createElement("option", {
      value: "c40"
    }, "Volvo S40"), /*#__PURE__*/_react.default.createElement("option", {
      value: "s60"
    }, "Volvo S60"), /*#__PURE__*/_react.default.createElement("option", {
      value: "s80"
    }, "Volvo S80"), /*#__PURE__*/_react.default.createElement("option", {
      value: "s90"
    }, "Volvo S90"));
  }

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: {
      flexDirection: 'row',
      paddingBottom: 20
    }
  }, /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Empty"
  }, /*#__PURE__*/_react.default.createElement(SelectInputWithValue, null)), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Filled"
  }, /*#__PURE__*/_react.default.createElement(SelectInputWithValue, {
    defaultValue: "entry"
  })), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Loading"
  }, /*#__PURE__*/_react.default.createElement(SelectInputWithValue, {
    loading: true,
    defaultValue: "entry"
  }))), /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: {
      flexDirection: 'row',
      paddingBottom: 20
    }
  }, /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Empty - Error"
  }, /*#__PURE__*/_react.default.createElement(SelectInputWithValue, {
    isValid: false,
    errorText: "I'm an error text"
  })), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Filled - Error"
  }, /*#__PURE__*/_react.default.createElement(SelectInputWithValue, {
    defaultValue: "entry",
    helperText: "I'm a helper text",
    errorText: "I'm an error text",
    isValid: false
  })), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Loading - Error"
  }, /*#__PURE__*/_react.default.createElement(SelectInputWithValue, {
    loading: true,
    isValid: false,
    defaultValue: "entry"
  }))), /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: {
      flexDirection: 'row',
      paddingBottom: 20
    }
  }, /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Empty - Disabled"
  }, /*#__PURE__*/_react.default.createElement(SelectInputWithValue, {
    disabled: true,
    helperText: "I'm a helper text"
  })), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Filled - Disabled"
  }, /*#__PURE__*/_react.default.createElement(SelectInputWithValue, {
    defaultValue: "entry",
    disabled: true
  })), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Loading - Disabled"
  }, /*#__PURE__*/_react.default.createElement(SelectInputWithValue, {
    loading: true,
    disabled: true,
    defaultValue: "entry"
  }))));
};

exports.Labelled = Labelled;

var Legacy = function Legacy() {
  function SelectInputWithValue(_ref3) {
    var _ref3$defaultValue = _ref3.defaultValue,
        defaultValue = _ref3$defaultValue === void 0 ? '' : _ref3$defaultValue,
        helperText = _ref3.helperText,
        errorText = _ref3.errorText,
        disabled = _ref3.disabled,
        loading = _ref3.loading,
        isValid = _ref3.isValid;
    return /*#__PURE__*/_react.default.createElement(_.SelectInput, {
      disabled: disabled,
      loading: loading,
      errorMessage: errorText,
      onChange: function onChange() {},
      description: helperText,
      placeholder: "e.g., Placeholder",
      value: defaultValue,
      isValid: isValid
    }, /*#__PURE__*/_react.default.createElement("option", {
      value: "entry"
    }, "Entry"), /*#__PURE__*/_react.default.createElement("option", {
      value: "c30",
      disabled: true
    }, "Volvo C30"), /*#__PURE__*/_react.default.createElement("option", {
      value: "c70"
    }, "Volvo C70"), /*#__PURE__*/_react.default.createElement("option", {
      value: "c90"
    }, "Volvo C90"), /*#__PURE__*/_react.default.createElement("option", {
      value: "c40"
    }, "Volvo S40"), /*#__PURE__*/_react.default.createElement("option", {
      value: "s60"
    }, "Volvo S60"), /*#__PURE__*/_react.default.createElement("option", {
      value: "s80"
    }, "Volvo S80"), /*#__PURE__*/_react.default.createElement("option", {
      value: "s90"
    }, "Volvo S90"));
  }

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: {
      flexDirection: 'row',
      paddingBottom: 20
    }
  }, /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Empty"
  }, /*#__PURE__*/_react.default.createElement(SelectInputWithValue, null)), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Filled"
  }, /*#__PURE__*/_react.default.createElement(SelectInputWithValue, {
    defaultValue: "entry"
  })), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Loading"
  }, /*#__PURE__*/_react.default.createElement(SelectInputWithValue, {
    loading: true,
    defaultValue: "entry"
  }))), /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: {
      flexDirection: 'row',
      paddingBottom: 20
    }
  }, /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Empty - Error"
  }, /*#__PURE__*/_react.default.createElement(SelectInputWithValue, {
    isValid: false,
    errorText: "I'm an error"
  })), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Filled - Error"
  }, /*#__PURE__*/_react.default.createElement(SelectInputWithValue, {
    defaultValue: "entry",
    isValid: false,
    helperText: "I'm a helper text",
    errorText: "I'm an error text"
  })), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Loading - Error"
  }, /*#__PURE__*/_react.default.createElement(SelectInputWithValue, {
    loading: true,
    isValid: false,
    defaultValue: "entry"
  }))), /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: {
      flexDirection: 'row',
      paddingBottom: 20
    }
  }, /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Empty - Disabled"
  }, /*#__PURE__*/_react.default.createElement(SelectInputWithValue, {
    disabled: true,
    helperText: "with helper"
  })), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Filled - Disabled"
  }, /*#__PURE__*/_react.default.createElement(SelectInputWithValue, {
    defaultValue: "entry",
    disabled: true
  })), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Loading - Disabled"
  }, /*#__PURE__*/_react.default.createElement(SelectInputWithValue, {
    loading: true,
    disabled: true,
    defaultValue: "entry"
  }))));
};

exports.Legacy = Legacy;

var OptGroups = function OptGroups() {
  var Example = /*#__PURE__*/function (_React$Component) {
    (0, _inherits2.default)(Example, _React$Component);

    var _super = _createSuper(Example);

    function Example() {
      var _this;

      (0, _classCallCheck2.default)(this, Example);

      for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
        args[_key] = arguments[_key];
      }

      _this = _super.call.apply(_super, [this].concat(args));
      (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "state", {
        value: 's80'
      });
      (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "handleChange", function (e) {
        _this.setState({
          value: e.target.value
        });
      });
      return _this;
    }

    (0, _createClass2.default)(Example, [{
      key: "render",
      value: function render() {
        return /*#__PURE__*/_react.default.createElement(_.SelectInput, {
          value: this.state.value,
          onChange: this.handleChange
        }, /*#__PURE__*/_react.default.createElement("optgroup", {
          label: "Berlin"
        }, /*#__PURE__*/_react.default.createElement("option", {
          value: "c70"
        }, "Volvo C70"), /*#__PURE__*/_react.default.createElement("option", {
          value: "c90"
        }, "Volvo C90")), /*#__PURE__*/_react.default.createElement("optgroup", {
          label: "G\xF6teborg"
        }, /*#__PURE__*/_react.default.createElement("option", {
          value: "s80"
        }, "Volvo S80"), /*#__PURE__*/_react.default.createElement("option", {
          value: "s90"
        }, "Volvo S90")));
      }
    }]);
    return Example;
  }(_react.default.Component);

  return /*#__PURE__*/_react.default.createElement(Example, null);
};

exports.OptGroups = OptGroups;
