// vcc-ui/es/components/radioGroup/stories.js
import _slicedToArray from "@babel/runtime/helpers/slicedToArray";
import React, { useState } from 'react';
import { Radio } from '../radio';
import { View } from '../view';
import { RadioGroup } from './';
export default {
  title: 'RadioGroup'
};
export var Default = function Default() {
  var _useState = useState(),
      _useState2 = _slicedToArray(_useState, 2),
      gender = _useState2[0],
      setGender = _useState2[1];

  var isValid = gender !== undefined;
  return /*#__PURE__*/React.createElement(RadioGroup, {
    legend: "Gender",
    name: "gender",
    value: gender,
    onChange: function onChange(e) {
      return setGender(e.target.value);
    },
    description: "Select your gender.",
    errorMessage: !isValid && 'Please select your gender.',
    isValid: isValid
  }, /*#__PURE__*/React.createElement(View, {
    spacing: 0.5
  }, /*#__PURE__*/React.createElement(Radio, {
    value: "male",
    label: "Male"
  }), /*#__PURE__*/React.createElement(Radio, {
    value: "female",
    label: "Female"
  }), /*#__PURE__*/React.createElement(Radio, {
    value: "diverse",
    label: "Diverse"
  })));
};
