// vcc-ui/es/components/slider/stories.js
import React from 'react';
import { Slider } from './';
export default {
  title: 'Slider'
};
export var FiveSteps = function FiveSteps() {
  return /*#__PURE__*/React.createElement(Slider, {
    onChange: function onChange(value) {
      // eslint-disable-next-line no-console
      console.log(value);
    },
    initialValue: 1,
    step: 1,
    minValue: 1,
    maxValue: 5
  });
};
export var Fibonacci = function Fibonacci() {
  return /*#__PURE__*/React.createElement(Slider, {
    initialValue: 1,
    onChange: function onChange(value) {
      // eslint-disable-next-line no-console
      console.log(value);
    },
    valueList: [0, 1, 2, 3, 5, 8, 13, 21, 34, 66, 89, 144]
  });
};
export var Disabled = function Disabled() {
  return /*#__PURE__*/React.createElement(Slider, {
    disabled: true,
    onChange: function onChange(value) {
      // eslint-disable-next-line no-console
      console.log(value);
    },
    initialValue: 3,
    step: 1,
    minValue: 1,
    maxValue: 5
  });
};
