// vcc-ui/es/components/tab-nav/stories.js
import React from 'react';
import { TabNav } from './';
import { TabNavItem } from '../tab-nav-item';
import { Block } from '../block';
export default {
  title: 'TabNav'
};
export var Default = function Default() {
  return /*#__PURE__*/React.createElement(Block, null, /*#__PURE__*/React.createElement(TabNav, {
    textAlign: "center"
  }, /*#__PURE__*/React.createElement(TabNavItem, {
    isActive: true
  }, "Crossover/Wagon"), /*#__PURE__*/React.createElement(TabNavItem, null, "Sedan"), /*#__PURE__*/React.createElement(TabNavItem, null, "Hybrid")));
};
export var ItemsAsLinks = function ItemsAsLinks() {
  return /*#__PURE__*/React.createElement(Block, null, /*#__PURE__*/React.createElement(TabNav, {
    textAlign: "center"
  }, /*#__PURE__*/React.createElement(TabNavItem, {
    isActive: true,
    href: "somelink"
  }, "Crossover/Wagon"), /*#__PURE__*/React.createElement(TabNavItem, {
    href: "somelink"
  }, "Sedan"), /*#__PURE__*/React.createElement(TabNavItem, {
    href: "somelink"
  }, "Hybrid")));
};
export var WithBackLink = function WithBackLink() {
  return /*#__PURE__*/React.createElement(Block, null, /*#__PURE__*/React.createElement(TabNav, {
    textAlign: "center",
    backButton: {
      text: 'Go back',
      href: '/href'
    },
    showBackButtonOn: ['s', 'm', 'l']
  }, /*#__PURE__*/React.createElement(TabNavItem, {
    isActive: true
  }, "Crossover/Wagon"), /*#__PURE__*/React.createElement(TabNavItem, null, "Sedan"), /*#__PURE__*/React.createElement(TabNavItem, null, "Hybrid")));
};
export var WithBackButton = function WithBackButton() {
  return /*#__PURE__*/React.createElement(Block, null, /*#__PURE__*/React.createElement(TabNav, {
    textAlign: "center",
    backButton: {
      text: 'Go back',
      clickHandler: function clickHandler(e) {
        e.preventDefault();
      }
    },
    showBackButtonOn: ['s', 'm', 'l']
  }, /*#__PURE__*/React.createElement(TabNavItem, {
    isActive: true
  }, "Crossover/Wagon"), /*#__PURE__*/React.createElement(TabNavItem, null, "Sedan"), /*#__PURE__*/React.createElement(TabNavItem, null, "Hybrid")));
};
export var WithStickyChildren = function WithStickyChildren() {
  return /*#__PURE__*/React.createElement(TabNav, {
    textAlign: "center",
    stickyChildrenLeft: /*#__PURE__*/React.createElement(TabNavItem, {
      extend: {
        marginLeft: 20,
        marginRight: 0
      }
    }, "Menu")
  }, /*#__PURE__*/React.createElement(TabNavItem, {
    isActive: true
  }, "Crossover/Wagon"), /*#__PURE__*/React.createElement(TabNavItem, null, "Sedan"), /*#__PURE__*/React.createElement(TabNavItem, null, "Hybrid"));
};
export var WithBackButtonStickyChildren = function WithBackButtonStickyChildren() {
  return /*#__PURE__*/React.createElement(TabNav, {
    textAlign: "center",
    backButton: {
      text: 'Go Back',
      clickHandler: function clickHandler(e) {
        e.preventDefault();
      }
    },
    stickyChildrenLeft: /*#__PURE__*/React.createElement(TabNavItem, {
      extend: {
        marginLeft: 10,
        marginRight: 0
      }
    }, "Menu")
  }, /*#__PURE__*/React.createElement(TabNavItem, {
    isActive: true
  }, "Crossover/Wagon"), /*#__PURE__*/React.createElement(TabNavItem, null, "Sedan"), /*#__PURE__*/React.createElement(TabNavItem, null, "Hybrid"));
};
export var BackButtonAndStickyChildren = function BackButtonAndStickyChildren() {
  return /*#__PURE__*/React.createElement(TabNav, {
    textAlign: "center",
    backButton: {
      text: 'Go Back',
      clickHandler: function clickHandler(e) {
        e.preventDefault();
      }
    },
    stickyChildrenLeft: /*#__PURE__*/React.createElement(TabNavItem, {
      extend: {
        marginLeft: 10,
        marginRight: 0
      }
    }, "Left"),
    stickyChildrenRight: /*#__PURE__*/React.createElement(TabNavItem, {
      extend: {
        marginLeft: 20
      }
    }, "Right")
  }, /*#__PURE__*/React.createElement(TabNavItem, {
    isActive: true
  }, "Crossover/Wagon"), /*#__PURE__*/React.createElement(TabNavItem, null, "Sedan"), /*#__PURE__*/React.createElement(TabNavItem, null, "Hybrid"));
};
