// vcc-ui/lib/components/text/stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Responsive = exports.Nested = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = require("../../");

var _default = {
  title: 'Text'
};
exports.default = _default;
var TEXT = 'A Quick Brown Fox Jumps Over The Lazy Dog';

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "ootah"
  }, "Core Text Styles"), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 4
  }), /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Headings"), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Peary"), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "peary"
  }, TEXT)), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Cook"), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "cook"
  }, TEXT)), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Ootah"), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "ootah"
  }, TEXT)), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Ootah - Emphasis"), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "ootah",
    subStyle: "emphasis"
  }, TEXT)), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 4
  }), /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Sub-Headings"), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Hillary"), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "hillary"
  }, TEXT)), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Hillary - Inline Link"), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "hillary",
    subStyle: "inline-link"
  }, TEXT)), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Hillary - Emphasis"), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "hillary",
    subStyle: "emphasis"
  }, TEXT)), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 4
  }), /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Body"), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Columbus"), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "columbus"
  }, TEXT)), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Columbus - Inline Link"), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "columbus",
    subStyle: "inline-link"
  }, TEXT)), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Columbus - Emphasis"), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "columbus",
    subStyle: "emphasis"
  }, TEXT)), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 4
  }), /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Micro"), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Bates"), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "bates"
  }, TEXT)), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Bates - Inline Link"), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "bates",
    subStyle: "inline-link"
  }, TEXT)), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Bates - Emphasis"), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "bates",
    subStyle: "emphasis"
  }, TEXT)), "=======", /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 4
  }), /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Sub-Headings"), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Hillary"), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "hillary"
  }, TEXT)), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Hillary - Inline Link"), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "hillary",
    href: "",
    subStyle: "inline-link"
  }, TEXT)), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Hillary - Emphasis"), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "hillary",
    subStyle: "emphasis"
  }, TEXT)), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 4
  }), /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Body"), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Columbus"), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "columbus"
  }, TEXT)), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Columbus - Inline Link"), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "columbus",
    href: "",
    subStyle: "inline-link"
  }, TEXT)), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Columbus - Emphasis"), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "columbus",
    subStyle: "emphasis"
  }, TEXT)), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 4
  }), /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Micro"), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Bates"), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "bates"
  }, TEXT)), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Bates - Inline Link"), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "bates",
    href: "",
    subStyle: "inline-link"
  }, TEXT)), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Bates - Emphasis"), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "bates",
    subStyle: "emphasis"
  }, TEXT)), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 4
  }), /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Action Text"), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Amundsen"), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "amundsen"
  }, TEXT)), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 4
  }), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "ootah"
  }, "Outlier Text Styles"), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 4
  }), /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Micro"), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Kelly"), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "kelly"
  }, TEXT)), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Kelly - Inline Link"), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "kelly",
    href: "",
    subStyle: "inline-link"
  }, TEXT)), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Kelly - Emphasis"), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "kelly",
    subStyle: "emphasis"
  }, TEXT)), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 4
  }), /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Legal"), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Meir"), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "meir"
  }, TEXT)), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 4
  }), /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Brand Statement"), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "Yang"), /*#__PURE__*/_react.default.createElement(_.Text, {
    as: "h1",
    variant: "yang"
  }, TEXT)));
};

exports.Default = Default;

var Nested = function Nested() {
  return /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "hillary"
  }, "Hillary with", ' ', /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "ootah",
    subStyle: "emphasis"
  }, "nested Ootah emphasis")), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "cook"
  }, "Hillary with ", /*#__PURE__*/_react.default.createElement(_.Text, {
    extend: {
      fontStyle: 'italic'
    }
  }, "nested extends")), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: "ootah",
    subStyle: "emphasis"
  }, "Ootah emphasis with ", /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "standard"
  }, "nested standard")), /*#__PURE__*/_react.default.createElement(_.Spacer, {
    size: 2
  }), /*#__PURE__*/_react.default.createElement(_.Text, null, "Default (Columbus) with", /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "emphasis"
  }, "nested emphasis"), "and an", ' ', /*#__PURE__*/_react.default.createElement(_.Text, {
    subStyle: "inline-link",
    href: "#"
  }, "inline link")));
};

exports.Nested = Nested;

var Responsive = function Responsive() {
  return /*#__PURE__*/_react.default.createElement(_.Flex, null, /*#__PURE__*/_react.default.createElement(_.Text, {
    variant: ['hillary', 'columbus'],
    subStyle: ['standard', 'emphasis', 'standard', 'emphasis']
  }, "Responsive Text"));
};

exports.Responsive = Responsive;
