// vcc-ui/lib/components/text/utils.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.stripFlowElement = stripFlowElement;
exports.arrayifyResponsiveValue = arrayifyResponsiveValue;
exports.populateResponsiveArray = populateResponsiveArray;
exports.normalizeResponsiveValue = normalizeResponsiveValue;
exports.getBaseStyle = getBaseStyle;
exports.getStylesForEachBreakpoint = getStylesForEachBreakpoint;
exports.mergeStyles = mergeStyles;
exports.getRenderedElement = getRenderedElement;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _typeof2 = _interopRequireDefault(require("@babel/runtime/helpers/typeof"));

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function stripFlowElement(el) {
  return /^(p|h[1-6])$/.test(el) ? 'span' : el;
} // TODO: can we refactor this to not look that ugly?


function arrayifyResponsiveValue(value) {
  if ((0, _typeof2.default)(value) === 'object') {
    if (!Array.isArray(value)) {
      var _defaultValue = value.default,
          _fromM = value.fromM,
          _fromL = value.fromL,
          _fromXL = value.fromXL;
      return [_defaultValue, _fromM, _fromL, _fromXL];
    }

    var _value = (0, _slicedToArray2.default)(value, 4),
        defaultValue = _value[0],
        fromM = _value[1],
        fromL = _value[2],
        fromXL = _value[3];

    return [defaultValue, fromM, fromL, fromXL];
  }

  return [value, value, value, value];
}

function populateResponsiveArray(value) {
  return value.reduce(function (filled, item, index) {
    return [].concat((0, _toConsumableArray2.default)(filled), [item || filled[index - 1]]);
  }, []);
}

function normalizeResponsiveValue(value) {
  return populateResponsiveArray(arrayifyResponsiveValue(value));
}

function getBaseStyle() {
  var style = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  // eslint-disable-next-line
  var fromM = style.fromM,
      fromL = style.fromL,
      fromXL = style.fromXL,
      baseStyle = (0, _objectWithoutProperties2.default)(style, ["fromM", "fromL", "fromXL"]);
  return baseStyle;
}

var breakpoints = ['fromM', 'fromL', 'fromXL'];

function getStylesForEachBreakpoint(styles, breakpoint) {
  var breakpointSpan = breakpoints.indexOf(breakpoint) + 1;
  return breakpoints.slice(0, breakpointSpan).reduce(function (mergedStyles, breakpoint) {
    if (styles[breakpoint]) {
      return _objectSpread(_objectSpread({}, mergedStyles), styles[breakpoint]);
    }

    return mergedStyles;
  }, {});
}

function mergeStyles(_ref, subStyle, breakpoint) {
  var styles = _ref.standard.styles,
      subStyles = (0, _objectWithoutProperties2.default)(_ref, ["standard"]);
  var baseStyle = getBaseStyle(styles);
  var baseSubStyle = getBaseStyle(subStyles[subStyle] ? subStyles[subStyle].styles : {});
  return _objectSpread(_objectSpread(_objectSpread({}, baseStyle), baseSubStyle), getStylesForEachBreakpoint(styles, breakpoint));
}

function getRenderedElement(isRootElement, as, variant, subStyle, theme) {
  var defaultElement = as || typeof variant === 'string' && typeof subStyle === 'string' && theme.typeScale[variant][subStyle] && theme.typeScale[variant][subStyle].element || 'p';
  return isRootElement ? defaultElement : stripFlowElement(defaultElement);
}
