// vcc-ui/lib/components/nav/stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.HideOnScroll = exports.Sticky = exports.Default = exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _fixtures = require("../../fixtures");

var _ = require("./");

var _flex = require("../flex");

var _navItem = require("../nav-item");

var _block = require("../block");

var _logo = require("../logo");

var _default = {
  title: 'Nav'
};
exports.default = _default;

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement(_block.Block, null, /*#__PURE__*/_react.default.createElement(_.Nav, null, /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: function extend(_ref) {
      var theme = _ref.theme;
      return (0, _defineProperty2.default)({}, theme.breakpoints.fromL, {
        flexDirection: 'row'
      });
    }
  }, /*#__PURE__*/_react.default.createElement(_navItem.NavItem, null, "This is Volvo"), /*#__PURE__*/_react.default.createElement(_navItem.NavItem, {
    isActive: true
  }, "Build & Price"), /*#__PURE__*/_react.default.createElement(_navItem.NavItem, {
    isDropdown: true
  }, "Cars"), /*#__PURE__*/_react.default.createElement(_navItem.NavItem, {
    isDropdown: true,
    isActive: true
  }, "Bicycles"))), /*#__PURE__*/_react.default.createElement(_block.Block, {
    extend: {
      marginTop: '20px',
      fontSize: '48px',
      width: '320px'
    }
  }, _fixtures.loremIpsum));
};

exports.Default = Default;

var Sticky = function Sticky() {
  return /*#__PURE__*/_react.default.createElement(_block.Block, null, /*#__PURE__*/_react.default.createElement(_.Nav, {
    sticky: true
  }, /*#__PURE__*/_react.default.createElement(_block.Block, {
    extend: {
      padding: '20px'
    }
  }, /*#__PURE__*/_react.default.createElement(_logo.Logo, {
    type: "square"
  }))), /*#__PURE__*/_react.default.createElement(_block.Block, {
    extend: {
      marginTop: '100px',
      fontSize: '48px',
      width: '320px'
    }
  }, _fixtures.loremIpsum));
};

exports.Sticky = Sticky;

var HideOnScroll = function HideOnScroll() {
  return /*#__PURE__*/_react.default.createElement(_block.Block, null, /*#__PURE__*/_react.default.createElement(_.Nav, {
    sticky: true,
    hideOnScroll: true
  }, /*#__PURE__*/_react.default.createElement(_block.Block, {
    extend: {
      padding: '20px'
    }
  }, /*#__PURE__*/_react.default.createElement(_logo.Logo, {
    type: "square"
  }))), /*#__PURE__*/_react.default.createElement(_block.Block, {
    extend: {
      marginTop: '100px',
      fontSize: '48px',
      width: '320px'
    }
  }, _fixtures.loremIpsum));
};

exports.HideOnScroll = HideOnScroll;
