// vcc-ui/lib/components/checkbox/stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Container = exports.WithLink = exports.Default = exports.default = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _ = require(".");

var _flex = require("../flex");

var _text = require("../text");

var _spacer = require("../spacer");

var _view = require("../view");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var _default = {
  title: 'Checkbox'
};
exports.default = _default;

var DisplayWrapper = function DisplayWrapper(_ref) {
  var name = _ref.name,
      children = _ref.children;
  return /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: {
      padding: 20,
      paddingRight: 40,
      minWidth: 200
    }
  }, /*#__PURE__*/_react.default.createElement(_text.Text, null, name), /*#__PURE__*/_react.default.createElement(_spacer.Spacer, {
    size: 0.5
  }), children);
};

var Default = function Default() {
  function CheckboxWithValue(_ref2) {
    var _ref2$defaultValue = _ref2.defaultValue,
        defaultValue = _ref2$defaultValue === void 0 ? false : _ref2$defaultValue,
        disabled = _ref2.disabled,
        isValid = _ref2.isValid,
        errorMessage = _ref2.errorMessage,
        description = _ref2.description;
    return /*#__PURE__*/_react.default.createElement(_.Checkbox, {
      disabled: disabled,
      onChange: function onChange() {},
      label: "Checkbox",
      checked: defaultValue,
      errorMessage: errorMessage,
      description: description,
      isValid: isValid
    });
  }

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: {
      flexDirection: 'row',
      paddingBottom: 20
    }
  }, /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Unchecked"
  }, /*#__PURE__*/_react.default.createElement(CheckboxWithValue, {
    description: "with helper text"
  })), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Checked"
  }, /*#__PURE__*/_react.default.createElement(CheckboxWithValue, {
    defaultValue: true
  })), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Error"
  }, /*#__PURE__*/_react.default.createElement(CheckboxWithValue, {
    isValid: false,
    errorMessage: "with error text",
    description: "with helper text"
  }))), /*#__PURE__*/_react.default.createElement(_flex.Flex, {
    extend: {
      flexDirection: 'row',
      paddingBottom: 20
    }
  }, /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Unchecked - Disabled"
  }, /*#__PURE__*/_react.default.createElement(CheckboxWithValue, {
    disabled: true,
    description: "with helper text"
  })), /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Checked - Disabled"
  }, /*#__PURE__*/_react.default.createElement(CheckboxWithValue, {
    disabled: true,
    defaultValue: true
  }))));
};

exports.Default = Default;

var WithLink = function WithLink() {
  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2.default)(_useState, 2),
      checked = _useState2[0],
      setChecked = _useState2[1];

  return /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Checkbox with a link"
  }, /*#__PURE__*/_react.default.createElement(_.Checkbox, {
    onChange: function onChange() {
      return setChecked(!checked);
    },
    label: /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, "I accept the", ' ', /*#__PURE__*/_react.default.createElement("a", {
      href: "/",
      onClick: function onClick(e) {
        e.stopPropagation();
        e.preventDefault();
      }
    }, "terms of service."), ' ', "This is a very long label to test mobile layout as well."),
    checked: checked
  }));
};

exports.WithLink = WithLink;

var Container = function Container() {
  var _useState3 = (0, _react.useState)(false),
      _useState4 = (0, _slicedToArray2.default)(_useState3, 2),
      checked = _useState4[0],
      setChecked = _useState4[1];

  return /*#__PURE__*/_react.default.createElement(DisplayWrapper, {
    name: "Checkbox within a direction=row container"
  }, /*#__PURE__*/_react.default.createElement(_view.View, {
    direction: "row",
    width: 200,
    bg: "rgb(240, 240, 240)",
    padding: 1
  }, /*#__PURE__*/_react.default.createElement(_.Checkbox, {
    onChange: function onChange() {
      return setChecked(!checked);
    },
    label: "I am a long label and should wrap within my parent container",
    checked: checked
  })));
};

exports.Container = Container;
