// vcc-ui/lib/prop-types.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.responsiveStringProp = exports.responsiveNumberProp = exports.responsiveProp = exports.ruleType = exports.oneOf = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

/**
	We need to create our own version of this, as it's
	not possible to combine out-of-the-box PropTypes
	with custom function validators.
*/
var oneOf = function oneOf(validTypes) {
  return function (props, propName, componentName) {
    if (validTypes.indexOf(props[propName]) === -1) {
      return new Error("Invalid prop supplied to '".concat(componentName, "'. Must be one of ['").concat(validTypes.join("', '"), "']."));
    }
  };
};

exports.oneOf = oneOf;

var ruleType = _propTypes.default.oneOfType([_propTypes.default.object, _propTypes.default.func]);

exports.ruleType = ruleType;

var responsiveProp = _propTypes.default.oneOfType([_propTypes.default.arrayOf(_propTypes.default.oneOfType([_propTypes.default.number, _propTypes.default.string])), _propTypes.default.object, _propTypes.default.string, _propTypes.default.number]);

exports.responsiveProp = responsiveProp;

var responsiveNumberProp = _propTypes.default.oneOfType([_propTypes.default.arrayOf(_propTypes.default.number), _propTypes.default.object, _propTypes.default.number]);

exports.responsiveNumberProp = responsiveNumberProp;

var responsiveStringProp = _propTypes.default.oneOfType([_propTypes.default.arrayOf(_propTypes.default.string), _propTypes.default.object, _propTypes.default.string]);

exports.responsiveStringProp = responsiveStringProp;
