// vcc-ui/lib/components/block/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Block = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _reactFela = require("react-fela");

var style = {
  display: 'block'
};

var Block = /*#__PURE__*/_react.default.forwardRef(function (_ref, ref) {
  var children = _ref.children,
      className = _ref.className,
      _ref$as = _ref.as,
      As = _ref$as === void 0 ? 'div' : _ref$as,
      extend = _ref.extend,
      props = (0, _objectWithoutProperties2.default)(_ref, ["children", "className", "as", "extend"]);

  var _useFela = (0, _reactFela.useFela)(props),
      css = _useFela.css;

  return /*#__PURE__*/_react.default.createElement(As, (0, _extends2.default)({}, props, {
    ref: ref,
    className: css(style, extend) + (className ? ' ' + className : '')
  }), children);
});

exports.Block = Block;
Block.displayName = 'Block';
Block.propTypes = {
  /** Any valid React element, function, or a string specifying a name for an HTML element */
  as: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.element, _propTypes.default.elementType, _propTypes.default.func]),

  /** An object containing valid CSS style declarations */
  extend: _propTypes.default.oneOfType([_propTypes.default.object, _propTypes.default.func, _propTypes.default.array]),

  /** A JSX node */
  children: _propTypes.default.node,

  /** Add an additional custom className to element. Warning: make sure it doesn't collide with the classNames being generated for the atomic CSS */
  className: _propTypes.default.string
};
