// vcc-ui/lib/components/tab-nav/stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.BackButtonAndStickyChildren = exports.WithBackButtonStickyChildren = exports.WithStickyChildren = exports.WithBackButton = exports.WithBackLink = exports.ItemsAsLinks = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = require("./");

var _tabNavItem = require("../tab-nav-item");

var _block = require("../block");

var _default = {
  title: 'TabNav'
};
exports.default = _default;

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement(_block.Block, null, /*#__PURE__*/_react.default.createElement(_.TabNav, {
    textAlign: "center"
  }, /*#__PURE__*/_react.default.createElement(_tabNavItem.TabNavItem, {
    isActive: true
  }, "Crossover/Wagon"), /*#__PURE__*/_react.default.createElement(_tabNavItem.TabNavItem, null, "Sedan"), /*#__PURE__*/_react.default.createElement(_tabNavItem.TabNavItem, null, "Hybrid")));
};

exports.Default = Default;

var ItemsAsLinks = function ItemsAsLinks() {
  return /*#__PURE__*/_react.default.createElement(_block.Block, null, /*#__PURE__*/_react.default.createElement(_.TabNav, {
    textAlign: "center"
  }, /*#__PURE__*/_react.default.createElement(_tabNavItem.TabNavItem, {
    isActive: true,
    href: "somelink"
  }, "Crossover/Wagon"), /*#__PURE__*/_react.default.createElement(_tabNavItem.TabNavItem, {
    href: "somelink"
  }, "Sedan"), /*#__PURE__*/_react.default.createElement(_tabNavItem.TabNavItem, {
    href: "somelink"
  }, "Hybrid")));
};

exports.ItemsAsLinks = ItemsAsLinks;

var WithBackLink = function WithBackLink() {
  return /*#__PURE__*/_react.default.createElement(_block.Block, null, /*#__PURE__*/_react.default.createElement(_.TabNav, {
    textAlign: "center",
    backButton: {
      text: 'Go back',
      href: '/href'
    },
    showBackButtonOn: ['s', 'm', 'l']
  }, /*#__PURE__*/_react.default.createElement(_tabNavItem.TabNavItem, {
    isActive: true
  }, "Crossover/Wagon"), /*#__PURE__*/_react.default.createElement(_tabNavItem.TabNavItem, null, "Sedan"), /*#__PURE__*/_react.default.createElement(_tabNavItem.TabNavItem, null, "Hybrid")));
};

exports.WithBackLink = WithBackLink;

var WithBackButton = function WithBackButton() {
  return /*#__PURE__*/_react.default.createElement(_block.Block, null, /*#__PURE__*/_react.default.createElement(_.TabNav, {
    textAlign: "center",
    backButton: {
      text: 'Go back',
      clickHandler: function clickHandler(e) {
        e.preventDefault();
      }
    },
    showBackButtonOn: ['s', 'm', 'l']
  }, /*#__PURE__*/_react.default.createElement(_tabNavItem.TabNavItem, {
    isActive: true
  }, "Crossover/Wagon"), /*#__PURE__*/_react.default.createElement(_tabNavItem.TabNavItem, null, "Sedan"), /*#__PURE__*/_react.default.createElement(_tabNavItem.TabNavItem, null, "Hybrid")));
};

exports.WithBackButton = WithBackButton;

var WithStickyChildren = function WithStickyChildren() {
  return /*#__PURE__*/_react.default.createElement(_.TabNav, {
    textAlign: "center",
    stickyChildrenLeft: /*#__PURE__*/_react.default.createElement(_tabNavItem.TabNavItem, {
      extend: {
        marginLeft: 20,
        marginRight: 0
      }
    }, "Menu")
  }, /*#__PURE__*/_react.default.createElement(_tabNavItem.TabNavItem, {
    isActive: true
  }, "Crossover/Wagon"), /*#__PURE__*/_react.default.createElement(_tabNavItem.TabNavItem, null, "Sedan"), /*#__PURE__*/_react.default.createElement(_tabNavItem.TabNavItem, null, "Hybrid"));
};

exports.WithStickyChildren = WithStickyChildren;

var WithBackButtonStickyChildren = function WithBackButtonStickyChildren() {
  return /*#__PURE__*/_react.default.createElement(_.TabNav, {
    textAlign: "center",
    backButton: {
      text: 'Go Back',
      clickHandler: function clickHandler(e) {
        e.preventDefault();
      }
    },
    stickyChildrenLeft: /*#__PURE__*/_react.default.createElement(_tabNavItem.TabNavItem, {
      extend: {
        marginLeft: 10,
        marginRight: 0
      }
    }, "Menu")
  }, /*#__PURE__*/_react.default.createElement(_tabNavItem.TabNavItem, {
    isActive: true
  }, "Crossover/Wagon"), /*#__PURE__*/_react.default.createElement(_tabNavItem.TabNavItem, null, "Sedan"), /*#__PURE__*/_react.default.createElement(_tabNavItem.TabNavItem, null, "Hybrid"));
};

exports.WithBackButtonStickyChildren = WithBackButtonStickyChildren;

var BackButtonAndStickyChildren = function BackButtonAndStickyChildren() {
  return /*#__PURE__*/_react.default.createElement(_.TabNav, {
    textAlign: "center",
    backButton: {
      text: 'Go Back',
      clickHandler: function clickHandler(e) {
        e.preventDefault();
      }
    },
    stickyChildrenLeft: /*#__PURE__*/_react.default.createElement(_tabNavItem.TabNavItem, {
      extend: {
        marginLeft: 10,
        marginRight: 0
      }
    }, "Left"),
    stickyChildrenRight: /*#__PURE__*/_react.default.createElement(_tabNavItem.TabNavItem, {
      extend: {
        marginLeft: 20
      }
    }, "Right")
  }, /*#__PURE__*/_react.default.createElement(_tabNavItem.TabNavItem, {
    isActive: true
  }, "Crossover/Wagon"), /*#__PURE__*/_react.default.createElement(_tabNavItem.TabNavItem, null, "Sedan"), /*#__PURE__*/_react.default.createElement(_tabNavItem.TabNavItem, null, "Hybrid"));
};

exports.BackButtonAndStickyChildren = BackButtonAndStickyChildren;
