// vcc-ui/lib/components/link/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Link = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactFela = require("react-fela");

var _click = require("../click");

var _inline = require("../inline");

var _helpers = require("../../helpers");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var Svg = function Svg(_ref) {
  var _ref$As = _ref.As,
      As = _ref$As === void 0 ? 'svg' : _ref$As,
      extend = _ref.extend,
      children = _ref.children,
      props = (0, _objectWithoutProperties2.default)(_ref, ["As", "extend", "children"]);

  var _useFela = (0, _reactFela.useFela)(props),
      css = _useFela.css;

  return /*#__PURE__*/_react.default.createElement(As, (0, _extends2.default)({
    className: css(extend)
  }, props), children);
};

var arrows = {
  left: 'M7 9.5l-4-4 4-4',
  right: 'M2 1.5l4 4-4 4',
  up: 'M1 6.5l4-4 4 4',
  down: 'M9 3.5l-4 4-4-4'
};

var Arrow = function Arrow(_ref2) {
  var arrowPath = _ref2.arrowPath,
      before = _ref2.before;
  return /*#__PURE__*/_react.default.createElement(Svg, {
    extend: _objectSpread(_objectSpread({}, before ? {
      marginRight: '0.5rem'
    } : {
      marginLeft: '0.5rem'
    }), {}, {
      top: '0.0625rem',
      position: 'relative'
    }),
    preserveAspectRatio: "xMinYMin meet",
    viewBox: "0 0 11 11",
    width: "0.6875rem",
    height: "0.6875rem",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, /*#__PURE__*/_react.default.createElement("path", {
    d: arrowPath,
    strokeWidth: "1.5",
    strokeLinecap: "round",
    strokeLinejoin: "round"
  }));
};

var Link = /*#__PURE__*/_react.default.forwardRef(function (_ref3, ref) {
  var children = _ref3.children,
      arrow = _ref3.arrow,
      props = (0, _objectWithoutProperties2.default)(_ref3, ["children", "arrow"]);

  var _useFela2 = (0, _reactFela.useFela)(),
      theme = _useFela2.theme;

  var rel = (0, _helpers.getLinkRel)(props);
  var resolveArrow = arrow;

  if (theme.direction === 'rtl') {
    if (arrow == 'right') {
      resolveArrow = 'left';
    }

    if (arrow == 'left') {
      resolveArrow = 'right';
    }
  }

  var arrowPath = arrows[resolveArrow];
  return /*#__PURE__*/_react.default.createElement(_click.Click, (0, _extends2.default)({
    ref: ref,
    rel: rel
  }, props, {
    extend: {
      fontSize: '0.875rem',
      fontWeight: 500,
      letterSpacing: '0.02em',
      padding: '12px 0',
      lineHeight: '1.375rem',
      fontFamily: theme.fontTypes.NOVUM,
      textTransform: 'uppercase',
      color: theme.tokens.linkForeground,
      stroke: theme.tokens.linkForeground,
      extend: [{
        condition: props.disabled,
        style: {
          opacity: 0.3
        }
      }, {
        condition: !props.disabled,
        style: {
          '&:hover': {
            color: theme.tokens.linkHoverForeground,
            stroke: theme.tokens.linkHoverForeground
          },
          '&:active': {
            color: theme.tokens.linkHoverForeground,
            stroke: theme.tokens.linkHoverForeground
          }
        }
      }]
    }
  }), arrow && arrow === 'left' && /*#__PURE__*/_react.default.createElement(_inline.Inline, {
    extend: {
      whiteSpace: 'nowrap'
    }
  }, /*#__PURE__*/_react.default.createElement(Arrow, {
    before: true,
    arrowPath: arrowPath
  }), "\uFEFF"), children, arrow && arrow !== 'left' && /*#__PURE__*/_react.default.createElement(_inline.Inline, {
    extend: {
      whiteSpace: 'nowrap'
    }
  }, "\uFEFF", /*#__PURE__*/_react.default.createElement(Arrow, {
    arrowPath: arrowPath
  })));
});

exports.Link = Link;
Link.displayName = 'Link';
Link.propTypes = {
  /** A JSX node */
  children: _propTypes.default.node,
  href: _propTypes.default.string,
  arrow: _propTypes.default.oneOf(['up', 'down', 'right', 'left']),

  /** Disable link interaction */
  disabled: _propTypes.default.bool
};
