// vcc-ui/lib/components/loading-bar/stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FixedToTop = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = require("./");

var _block = require("../block");

var _default = {
  title: 'LoadingBar'
};
exports.default = _default;

var Container = function Container(_ref) {
  var children = _ref.children;
  return /*#__PURE__*/_react.default.createElement(_block.Block, {
    extend: {
      padding: 20
    }
  }, /*#__PURE__*/_react.default.createElement(_block.Block, {
    extend: {
      padding: 40
    }
  }, children));
};

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement(Container, null, /*#__PURE__*/_react.default.createElement(_.LoadingBar, {
    isLoading: ('isLoading', true),
    transitionDuration: 1500
  }));
};

exports.Default = Default;

var FixedToTop = function FixedToTop() {
  return /*#__PURE__*/_react.default.createElement(_block.Block, {
    extend: {
      position: 'fixed',
      top: 0,
      left: 0,
      width: '100%'
    }
  }, /*#__PURE__*/_react.default.createElement(_.LoadingBar, {
    isLoading: ('isLoading', true)
  }));
};

exports.FixedToTop = FixedToTop;
